---
title: "SMB_HYBRID_DETECTIVE"
author: "Joe Gunn"
date: "7/10/2020"
output: html_document
---

## Libraries needed for analysis
```{r}
library(genepopedit)
library(parallelnewhybrid)
library(hybriddetective)
library(readxl)
library(tidyverse)
library(adegenet)
library(cowplot)
```

## Get top 200 diagnostic loci
```{r}
#Expore genepop data using genepopedit package - here, I am just seeing what the data tables look like with respect to the genepop edit package. I am then using the genepopedit pacakge to make sure that all of the genepop files are in the correct format for the hybriddetective getTopLoc() function

genepop_detective("../../raw_data/hybriddetective_data/hyb_detective_genepops/hsyc_canbar_white_parent_genepop_goodformat.txt", variable = "PopNum")
genepop_flatten("../../raw_data/hybriddetective_data/hyb_detective_genepops/hsyc_canbar_white_parent_genepop_goodformat.txt")

#create a dataframe to designate two different populations (Pure Neosho, Pure Northern)
pop_rename_hsyc_white <- data.frame(Opop = c("","","","","","","","","","","","","","","","","",""), Rename = c("HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

pop_rename_canbar_skia <- data.frame(Opop = c("","","","","","","","","","","","","",""), Rename = c("CANBAR","CANBAR","CANBAR","CANBAR","CANBAR","CANBAR","CANBAR","CANBAR","CANBAR","SKIA","SKIA","SKIA","SKIA","SKIA"))

pop_rename_hsyc_canbar_white <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","HSYCCANBAR","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))


#Designate the two different populations (Pure Neosho, Pure Northern)
subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/hsyc_white_parent_genepop.txt", pop_rename_hsyc_white, path = "/Users/joegunn/Desktop/hsyc_white_parent_genepop_goodformat.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/canbar_skia_parent_genepop.txt", pop_rename_canbar_skia, path = "/Users/joegunn/Desktop/canbar_skia_parent_genepop_goodformat.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/hsyc_canbar_white_parent_genepop.txt", pop_rename_hsyc_canbar_white, path = "/Users/joegunn/Desktop/hsyc_canbar_white_parent_genepop_goodformat.txt")



#Run command to extract the top 200 diagnositic SNPs (with highest Fst values) across pure Neosho and Northern populations. Both PGDSpider and PLINK needed to be installed on my local computer, and the respective paths to both executables are given in this function. 
```

## Run getTopLoc() funtion from hybriddetective to get 200 diagnostic, high linkage, high Fst SNPs. There must be a "/" after the directories that PLINK and PGDSpider are in!
```{r}
#Get panel SNP panel for HSYC and WHITE populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_format_genepops/hsyc_white_parent_genepop_goodformat.txt", panel.size = 200,
where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/", where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")

#Get panel SNP panel for CANBAR and SKIA populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_format_genepops/canbar_skia_parent_genepop_goodformat.txt", panel.size = 200,
where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/", where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")

#Get panel SNP panel for HSYC+CANBAR AND WHITE populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_format_genepops/hsyc_canbar_white_parent_genepop_goodformat.txt", panel.size = 200,
where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/", where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")
```

## Determine the SNP names for the 200 diagnostic SNPs discovered in the above analyses - this is necessary, because the hybriddetective workflow code just generates the number of the SNPs that were used to make the panel (SNP_1, 2, 3, etc.) rather than the actual SNP name.
```{r}
snp_metadata <- read_excel("../../metadata/smb_snp_metadata.xlsx")
snp_metadata <- snp_metadata %>%
  mutate(snp_number = factor(snp_number))

#Honey Creek and Sycamore Creek with the White River SNPs
hsyc_white_200_panel_snps <- read_excel("../../metadata/hsyc_white_200_panel.xlsx")
hsyc_white_200_panel_snps <- hsyc_white_200_panel_snps %>%
  mutate(snp_number = factor(snp_number))

hsyc_white_200_panel_snp_names <- merge(snp_metadata, hsyc_white_200_panel_snps, by = "snp_number")

#Honey Creek and Sycamore Creek with Skiatook Lake SNPs
hsyc_skia_200_panel_snps <- read_excel("../../metadata/hsyc_skia_200_panel.xlsx")
hsyc_skia_200_panel_snps <- hsyc_skia_200_panel_snps %>%
  mutate(snp_number = factor(snp_number))

hsyc_skia_200_panel_snp_names <- merge(snp_metadata, hsyc_skia_200_panel_snps, by = "snp_number")

#Write out tables with SNP names to be kept for hybrid analysis - these will be text files that can be later used in vcftools to subset vcfs on Lewis 
write.table(hsyc_white_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/metadata/hsyc_white_200_panel_snp_names.txt", sep="\t")

write.table(hsyc_skia_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/metadata/hsyc_skia_200_panel_snp_names.txt", sep="\t")
```

## HYBRIDLAB ANALYSIS
### In the program HYBRIDLAB, I am simulating 6 hybrid categories (P1, P2, F1, F2, BC1, BC2) for hybrid populations between HSYC and WHITE and between HSYC and SKIA. I am generating 100 simulated individuals for each category, including simulated parent individuals, even though I have empirical data for the parents. This will allow me to assess all differences or similarities with equal sample sizes. Simulated genotypes will be compared to the individual admixed populations from my study

## Format simulated data files from HYBRIDLAB 
```{r}
#Format genepop file for simulated HSYC and WHITE parent 1 and 2 
#pop_rename_hsyc_white_p1_p2 <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_P1_P2_sim.txt", pop_rename_hsyc_white_p1_p2, path = "/Users/joegunn/Desktop/hsyc_white_P1_P2_sim_2pop.txt")

#Format genepop file for simulated HSYC and WHITE F1 backcross to HSYC 
#pop_rename_hsyc_white_F1_HSYC <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_F1_HSYC_sim.txt", pop_rename_hsyc_white_F1_HSYC, path = "/Users/joegunn/Desktop/hsyc_white_F1_HSYC_sim_2pop.txt")

#Format genepop file for simulated HSYC and WHITE F1 backcross to WHITE 
#pop_rename_hsyc_white_F1_WHITE <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_F1_WHITE_sim.txt", pop_rename_hsyc_white_F1_WHITE, path = "/Users/joegunn/Desktop/hsyc_white_F1_WHITE_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA parent 1 and 2 
#pop_rename_hsyc_skia_p1_p2 <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_P1_P2_sim.txt", pop_rename_hsyc_skia_p1_p2, path = "/Users/joegunn/Desktop/hsyc_skia_P1_P2_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA F1 backcross to HSYC 
#pop_rename_hsyc_skia_F1_HSYC <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_F1_HSYC_sim.txt", pop_rename_hsyc_skia_F1_HSYC, path = "/Users/joegunn/Desktop/hsyc_skia_F1_HSYC_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA F1 backcross to SKIA 
#pop_rename_hsyc_skia_F1_SKIA <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_F1_SKIA_sim.txt", pop_rename_hsyc_skia_F1_SKIA, path = "/Users/joegunn/Desktop/hsyc_skia_F1_SKIA_sim_2pop.txt")
```

## Format all hybrid datafiles
```{r}
##WHITE RIVER ADMIXED POPS
#pop_rename_white_bayou <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("AT","AT","AT","AT","AT","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BAYOU_genepop.txt", pop_rename_white_bayou, path = "/Users/joegunn/Desktop/hsyc_white_BAYOU_genepop_correct.txt")


#pop_rename_white_bsug <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("MI","MI","MI","MI","MI","MI","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BSUG_genepop.txt", pop_rename_white_bsug, path = "/Users/joegunn/Desktop/hsyc_white_BSUG_genepop_correct.txt")


#pop_rename_white_buff <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("BC","BC","BC","BC","BC","BC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BUFF_genepop.txt", pop_rename_white_buff, path = "/Users/joegunn/Desktop/hsyc_white_BUFF_genepop_correct.txt")


#pop_rename_white_elk <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("ELK","ELK","ELK","ELK","ELK","ELK","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_ELK_genepop.txt", pop_rename_white_elk, path = "/Users/joegunn/Desktop/hsyc_white_ELK_genepop_correct.txt")


#pop_rename_white_spav <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("SPVW","SPVW","SPVW","SPVW","SPVW","SPVW","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_SPAV_genepop.txt", pop_rename_white_spav, path = "/Users/joegunn/Desktop/hsyc_white_SPAV_genepop_correct.txt")



#SKIATOOK LAKE ADMIXED POPS
#pop_rename_skia_baron <- data.frame(Opop = c("","","","","","","","","","","","","","","","","",""), Rename = c("BFORK","BFORK","BFORK","BFORK","BFORK","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_BARON_genepop.txt", pop_rename_skia_baron, path = "/Users/joegunn/Desktop/hsyc_skia_BARON_genepop_correct.txt")


#pop_rename_skia_caney <- data.frame(Opop = c("","","","","","","","","","","","","","","","","",""), Rename = c("CANEY","CANEY","CANEY","CANEY","CANEY","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_CANEY_genepop.txt", pop_rename_skia_caney, path = "/Users/joegunn/Desktop/hsyc_skia_CANEY_genepop_correct.txt")


#pop_rename_skia_illinois <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","",""), Rename = c("ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_ILLINOIS_genepop.txt", pop_rename_skia_illinois, path = "/Users/joegunn/Desktop/hsyc_skia_ILLINOIS_genepop_correct.txt")


#genepop_flatten("../../../../../../hsyc_skia_ILLINOIS_genepop_correct.txt")
#genepop_detective("../../../../../../hsyc_skia_ILLINOIS_genepop_correct.txt", variable = "PopNum")
```

## DAPC plotting for hybridlab simulations
### In order to convert the simulated genepop data into a genind formatted file for DAPC in adegenet, I had to create a matrix of genotypes, rather than using a standard genepop format (I think there is a way to use a genepop file strictly, but I couldn't figure it out). My genotype matrix consisted of one row per individual sample, and each column was a SNP (200 total). Genotypes were coded as ###### (six numbers in a row)
```{r}
## Run DAPC for HSYC and SKIA parents, along with admixed populations (BARON, CANEY, ILLINOIS)
hsyc_skia_all_admix_genepop <- read_excel("../../raw_data/genepop_data/hybrid_lab_simulations/hybrid_category_genotypes/hsyc_skia/HSYC_SKIA_all_hybrids_allADMIX_plusParent.xlsx")

## Create population vector
admix_pops_skia <- factor(c(rep("HSYC", times = 100), rep("SKIA", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("BC_HSYC", times = 100), rep("BC_SKIA", times = 100), rep("BARON", times = 5), rep("CANEY", times = 5), rep("ILLINOIS", times = 7), rep("HSYC_emp", times = 8), rep("SKIA_emp", times = 5)))

hsyc_skia_all_admix_genepop <- column_to_rownames(hsyc_skia_all_admix_genepop, "ind")
hsyc_skia_all_admix_genepop <- as.matrix(hsyc_skia_all_admix_genepop)

## Get genind file
hsyc_skia_all_admix_genepop_genind <- df2genind(hsyc_skia_all_admix_genepop, 
                                                ncode = 3, 
                                                pop = admix_pops_skia, 
                                                ploidy = 2)
## Scale genind file
hsyc_skia_all_admix_genepop_genind <- scaleGen(hsyc_skia_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

## Run DAPC
dapc_hsyc_skia_admix <- dapc(hsyc_skia_all_admix_genepop_genind, admix_pops_skia)

## PCs retained: 10
## Discriminant Functions retained: 3

scatter(dapc_hsyc_skia_admix)

## Run DAPC for HSYC and WHITE parents, along with admixed populations (SPAV, ELK, BSUG, BUFF, BAYOU)
hsyc_white_all_admix_genepop <- read_excel("../../raw_data/genepop_data/hybrid_lab_simulations/hybrid_category_genotypes/hsyc_white/HSYC_WHITE_all_hybrids_allADMIX_plusParent.xlsx")

## Create population vector
admix_pops_white <- factor(c(rep("HSYC_sim", times = 100), rep("WHITE_sim", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("BC_HSYC", times = 100), rep("BC_WHITE", times = 100), rep("BAYOU", times = 5), rep("BSUG", times = 6), rep("BUFF", times = 6), rep("ELK", times = 6), rep("SPVW", times = 6), rep("HSYC_emp", times = 8), rep("WHITE_emp", times = 10)))

hsyc_white_all_admix_genepop <- column_to_rownames(hsyc_white_all_admix_genepop, "ind")
hsyc_white_all_admix_genepop <- as.matrix(hsyc_white_all_admix_genepop)

## Get genind file
hsyc_white_all_admix_genepop_genind <- df2genind(hsyc_white_all_admix_genepop, 
                                                 ncode = 3, 
                                                 pop = admix_pops_white, 
                                                 ploidy = 2)
## Scale genind file
hsyc_white_all_admix_genepop_genind <- scaleGen(hsyc_white_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

# Run DAPC
dapc_hsyc_white_admix <- dapc(hsyc_white_all_admix_genepop_genind, admix_pops_white)

## PCs retained: 10
## Discriminant Functions retained: 3

scatter(dapc_hsyc_white_admix)

```

## Make plots
```{r}
## HSYC SKIA Plot
dapc_hsyc_skia_admix_LDs <- as.data.frame(dapc_hsyc_skia_admix$ind.coord)
dapc_hsyc_skia_admix_LDs <- cbind(data.frame(admix_pops_skia), dapc_hsyc_skia_admix_LDs)
colnames(dapc_hsyc_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_skia_admix_plot <- ggplot(dapc_hsyc_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "grey35","grey85", "firebrick3","grey70","grey50","grey15","deepskyblue","mediumpurple", "grey100", "navyblue")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20))

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_plot <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("lightgreen", "grey35","grey85", "sienna4", "lightyellow","lightsalmon","grey70","grey50","deepskyblue","grey15", "orchid1", "deeppink2", "grey100")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.text.x = element_blank()) +
   theme(axis.title.x = element_blank())


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybrid_detective/hyb_simulation_plots.pdf", width=9, height=12)

plot_grid(dapc_hsyc_white_admix_plot, dapc_hsyc_skia_admix_plot, nrow = 2, labels = c("a","b"), label_size = 27)

dev.off()
```

## BLACK AND WHITE plots
```{r}
## HSYC SKIA Plot
dapc_hsyc_skia_admix_LDs <- as.data.frame(dapc_hsyc_skia_admix$ind.coord)
dapc_hsyc_skia_admix_LDs <- cbind(data.frame(admix_pops), dapc_hsyc_skia_admix_LDs)
colnames(dapc_hsyc_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_skia_admix_plot_black <- ggplot(dapc_hsyc_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "grey44","grey100", "firebrick3","grey84","grey64","deepskyblue","mediumpurple","navyblue", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(legend.text = element_text(color = "white")) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_plot_black <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("lightgreen", "grey44","grey100", "sienna4", "lightyellow","lightsalmon","grey84","grey64","deepskyblue","orchid1", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20, color = "white")) +
   theme(axis.text = element_text(size = 20, color = "white")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybrid_detective/hyb_simulation_plots_black.pdf", width=9, height=12)

plot_grid(dapc_hsyc_white_admix_plot_black, dapc_hsyc_skia_admix_plot_black, nrow = 2, labels = c("a","b"), label_size = 27)

dev.off()
```
