---
title: "SMB_HYBRID_DETECTIVE"
author: "Joe Gunn"
date: "7/10/2020"
output: html_document
---

## Libraries needed for analysis
```{r}
library(genepopedit)
library(parallelnewhybrid)
library(hybriddetective)
library(readxl)
library(tidyverse)
library(adegenet)
library(cowplot)
library(pophelper)
```

## Prepare parent genepop datasets for PLINK (I used pgdspider on my computer to convert vcf files for only the parent populations, MIDARK and WHITE, and MIDARK and SKIA, into genepop files. The genepop files created in pgdspider do not have the correct population designations, so they need to be reformatted with the "subset_genepop_rename" code below).

## For each original vcf file (MIDARK and WHITE, MIDARK and SKIA), I randomized the order of the RAD tags listed (rows in the vcf file), because, based on my experience,  the getTopLoc() function below, which finds the most diagnostic snps in the dataset, considers each snp in order. So, I usually end up with only the first "diagnostic" snps in the whole vcf, which I do not think is a representative way to scan the whole dataset. Thus, I randomized rows so that each snp has a roughly equal probability of being selected as diagnostic. Rows were randomized in excel using the =RAND() function, which assigns each row a random number, and then organizing rows in ascending order based on the generated random number.
```{r}
#create a dataframe to designate two different populations (Pure Neosho, Pure Northern). This must be done to make sure the genepop comes out in the correct format
pop_rename_midark_white <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

pop_rename_midark_skia <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","SKIA","SKIA","SKIA","SKIA","SKIA"))

#Designate the two different populations (Pure Neosho, Pure Northern)
subset_genepop_rename("../../raw_data/hybriddetective_data/parent_genepops/MIDARK_WHITE_parent_genepop_snps_random.txt", pop_rename_midark_white, path = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_genepops/MIDARK_WHITE_edited_parent_genepop_snps_random.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/parent_genepops/MIDARK_SKIA_parent_genepop_snps_random.txt", pop_rename_midark_skia, path = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_genepops/MIDARK_SKIA_edited_parent_genepop_snps_random.txt")


#Explore genepop data using genepopedit package - here, I am just seeing what the data tables look like with respect to the genepop edit package. I am then using the genepopedit pacakge to make sure that all of the genepop files are in the correct format for the hybriddetective getTopLoc() function
genepop_flatten("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_genepops/MIDARK_SKIA_edited_parent_genepop_snps_random.txt")
```
## Get top 200 diagnostic loci

## Run getTopLoc() funtion from hybriddetective to get 200 diagnostic, high linkage, high Fst SNPs. There must be a "/" after the directories that PLINK and PGDSpider are in! Otherwise the code won't work and you'll be very confused.
```{r}
#Get panel SNP panel for MIDARK and WHITE populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_genepops/MIDARK_WHITE_edited_parent_genepop_snps_random.txt",
          panel.size = 200,
          where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/",
          where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")

#Get panel SNP panel for MIDARK and SKIA populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_genepops/MIDARK_SKIA_edited_parent_genepop_snps_random.txt", 
          panel.size = 200,
          where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/", 
          where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")
```
## Diagnostic Loci validation

### PLINK selects the 200 most diagnostic loci only based on approximately 1/2 of the pure parent samples to train the simulation alogrithm, and then generates a new genepop file that includes only the remaining (non-training) pure parent samples and their corresponding genoptypes at the same loci as a validation set. These validation samples can then be used to simulate 6 hybrid cateogries (Parent 1 (MIDARK), Parent 2 (SKIA or WHITE), F1, F2, BC1, and BC2), and the program NewHybrids can validate whether or not these loci were able to successfully identify hybrids based on the genotypes produced.

## Here I am simulating genotypes for 6 hybrid categories using the 200 diagnostic loci discovered in the PLINK validation samples for each parent dataset. The freqbasedsim_AlleleSample() function generates a number of individuals in each hybrid category based on allele frequencies

## Genotypes were simulated 3 separate times (S1, S2, S3), and for each set of simulations (unique genotype datasest), I performed 3 replicate hybrid generation simulations (R1, R2, R3). This way, I can assess deviations across simulations (S) AND within simulations (R).
```{r}
#Simulate hybrid genotypes based on the VALIDATION parent samples (the non-training parent samples - those not used to discover the 200 most informative, diagnostic loci)

#Simulate for MIDARK and WHITE
freqbasedsim_AlleleSample(GPD ="../../analysis/hybriddetective_analysis/parent_200_loci_validation_genepops/MIDARK_WHITE_edited_parent_genepop_snps_random_200_Loci_Panel.txt", 
                          NumSims = 3, 
                          NumReps = 3)

#Simulate for MIDARK and SKIA
freqbasedsim_AlleleSample(GPD ="../../analysis/hybriddetective_analysis/parent_200_loci_validation_genepops/MIDARK_SKIA_edited_parent_genepop_snps_random_200_Loci_Panel.txt", 
                          NumSims = 3, 
                          NumReps = 3)
```

## Validation of simulated data in NewHybrids using the parallelnewhybrids() package
```{r}

#Run parallelnewhybrids program to assign simulated genotypes to hybrid class (Bayesian Gibbs sampling algorithm). NewHybrids was installed on my local machine (MAC OS) to conduct the analysis
parallelnh_OSX("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/", 
               where.NH = "/Users/joegunn/Downloads/newhybrids/", 
               burnin = 100000, 
               sweeps = 500000)

#Run parallelnewhybrids program to assign simulated genotypes to hybrid class (Bayesian Gibbs sampling algorithm)
parallelnh_OSX("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/", 
               where.NH = "/Users/joegunn/Downloads/newhybrids/", 
               burnin = 100000, 
               sweeps = 500000)

#Check for NewHybrids model convergence - this code will generate a message that either says to re-check the results (the Bayesian threads did not converge) or that you are good to go (the threads successfully converged). Failure to converge results in most individuals being assigned erroneously to the "F2" hybrid class.

#Check for MIDARK and WHITE
nh_preCheckR(PreDir="/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/")

#Check for MIDARK and SKIA
nh_preCheckR(PreDir="/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/")
```

## Graph Validation Results
```{r}
#Graph results for MIDARK and WHITE

##SIMULATION 1 (S1): Replicates 1,2,3
midark_white_s1r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S1R1_PofZ.xlsx")

midark_white_s1r1_data_gather <- midark_white_s1r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s1r1_plot <- ggplot(midark_white_s1r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank())

midark_white_s1r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S1R2_PofZ.xlsx")

midark_white_s1r2_data_gather <- midark_white_s1r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s1r2_plot <- ggplot(midark_white_s1r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) +
   theme(axis.title.y = element_blank())

midark_white_s1r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S1R3_PofZ.xlsx")

midark_white_s1r3_data_gather <- midark_white_s1r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s1r3_plot <- ggplot(midark_white_s1r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.y = element_blank()) + 
   theme(axis.title.x = element_blank())


##SIMULATION 2 (S2): Replicates 1,2,3
midark_white_s2r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S2R1_PofZ.xlsx")

midark_white_s2r1_data_gather <- midark_white_s2r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s2r1_plot <- ggplot(midark_white_s2r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank())

midark_white_s2r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S2R2_PofZ.xlsx")

midark_white_s2r2_data_gather <- midark_white_s2r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s2r2_plot <- ggplot(midark_white_s2r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) +
   theme(axis.title.y = element_blank())

midark_white_s2r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S2R3_PofZ.xlsx")

midark_white_s2r3_data_gather <- midark_white_s2r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s2r3_plot <- ggplot(midark_white_s2r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.y = element_blank()) + 
   theme(axis.title.x = element_blank())


##SIMULATION 3 (S3): Replicates 1,2,3
midark_white_s3r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S3R1_PofZ.xlsx")

midark_white_s3r1_data_gather <- midark_white_s3r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s3r1_plot <- ggplot(midark_white_s3r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) 

midark_white_s3r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S3R2_PofZ.xlsx")

midark_white_s3r2_data_gather <- midark_white_s3r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s3r2_plot <- ggplot(midark_white_s3r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.y = element_blank())

midark_white_s3r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_white/NH.Results/PoZ_results/MIDARK_WHITE_S3R3_PofZ.xlsx")

midark_white_s3r3_data_gather <- midark_white_s3r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_white_s3r3_plot <- ggplot(midark_white_s3r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.y = element_blank())

#Plot MIDARK and SKIA Validation Results
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybriddetective_figures/midark_white_nh_validation_plots.pdf", width=12, height=12)

plot_grid(midark_white_s1r1_plot, 
          midark_white_s1r2_plot, 
          midark_white_s1r3_plot, 
          midark_white_s2r1_plot, 
          midark_white_s2r2_plot, 
          midark_white_s2r3_plot, 
          midark_white_s3r1_plot, 
          midark_white_s3r2_plot, 
          midark_white_s3r3_plot,
          nrow = 3,
          ncol = 3,
          labels = c("a","b","c","d","e","f","g","h","i"),
          label_y = 1.02)

dev.off()
#####################################################################
#####################################################################


#Graph results for MIDARK and SKIA

##SIMULATION 1 (S1): Replicates 1,2,3
midark_skia_s1r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S1R1_PofZ.xlsx")

midark_skia_s1r1_data_gather <- midark_skia_s1r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s1r1_plot <- ggplot(midark_skia_s1r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank())

midark_skia_s1r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S1R2_PofZ.xlsx")

midark_skia_s1r2_data_gather <- midark_skia_s1r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s1r2_plot <- ggplot(midark_skia_s1r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) + 
   theme(axis.title.y = element_blank())

midark_skia_s1r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S1R3_PofZ.xlsx")

midark_skia_s1r3_data_gather <- midark_skia_s1r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s1r3_plot <- ggplot(midark_skia_s1r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) + 
   theme(axis.title.y = element_blank())


##SIMULATION 2 (S2): Replicates 1,2,3
midark_skia_s2r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S2R1_PofZ.xlsx")

midark_skia_s2r1_data_gather <- midark_skia_s2r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s2r1_plot <- ggplot(midark_skia_s2r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank())

midark_skia_s2r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S2R2_PofZ.xlsx")

midark_skia_s2r2_data_gather <- midark_skia_s2r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s2r2_plot <- ggplot(midark_skia_s2r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) + 
   theme(axis.title.y = element_blank())

midark_skia_s2r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S2R3_PofZ.xlsx")

midark_skia_s2r3_data_gather <- midark_skia_s2r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s2r3_plot <- ggplot(midark_skia_s2r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) +
   theme(axis.title.x = element_blank()) + 
   theme(axis.title.y = element_blank())


##SIMULATION 3 (S3): Replicates 1,2,3
midark_skia_s3r1_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S3R1_PofZ.xlsx")

midark_skia_s3r1_data_gather <- midark_skia_s3r1_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s3r1_plot <- ggplot(midark_skia_s3r1_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100"))

midark_skia_s3r2_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S3R2_PofZ.xlsx")

midark_skia_s3r2_data_gather <- midark_skia_s3r2_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s3r2_plot <- ggplot(midark_skia_s3r2_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) + 
   theme(axis.title.y = element_blank())

midark_skia_s3r3_data <- read_excel("../../analysis/hybriddetective_analysis/newhybrid_validation_simulations_midark_skia/NH.Results/PoZ_results/MIDARK_SKIA_S3R3_PofZ.xlsx")

midark_skia_s3r3_data_gather <- midark_skia_s3r3_data %>%
   gather(PARENT1:BC_PARENT2, key = "hybrid_cat", value = "probability") %>%
   mutate(hybrid_id = factor(hybrid_id), hybrid_cat = factor(hybrid_cat))

midark_skia_s3r3_plot <- ggplot(midark_skia_s3r3_data_gather, aes(x = indiv_id, y = probability, fill = hybrid_id)) + 
   geom_bar(stat = "identity", color = "black", show.legend = F) +
   theme_set(theme_cowplot(12)) +
   labs(x = "Simulated Individual", y = "Cumulative Probability of Hybrid ID") +
   scale_fill_manual(values = c("grey18","grey90","grey35","grey60","grey5","grey100")) + 
   theme(axis.title.y = element_blank())

#Plot MIDARK and SKIA Validation Results
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybriddetective_figures/midark_skia_nh_validation_plots.pdf", width=12, height=12)

plot_grid(midark_skia_s1r1_plot, 
          midark_skia_s1r2_plot, 
          midark_skia_s1r3_plot, 
          midark_skia_s2r1_plot, 
          midark_skia_s2r2_plot, 
          midark_skia_s2r3_plot, 
          midark_skia_s3r1_plot, 
          midark_skia_s3r2_plot, 
          midark_skia_s3r3_plot,
          nrow = 3,
          ncol = 3,
          labels = c("a","b","c","d","e","f","g","h","i"),
          label_y = 1.02)

dev.off()
```

## Get SNP IDs for 200 diagnostic SNPs to be used for revised, full diagnostic genepop file
```{r}
#MIDARK and WHITE 200 diagnostic SNP names
midark_white_random_snps <- read_excel("../../analysis/hybriddetective_analysis/snps_randomized_names/MIDARK_WHITE_random_snp_names.xlsx")

midark_white_200_panel_snps <- read_excel("../../analysis/hybriddetective_analysis/snps_randomized_names/MIDARK_WHITE_random_200_loci_snp_names.xlsx")

midark_white_200_panel_snps <- midark_white_200_panel_snps %>%
  mutate(snp_num = factor(snp_num))

midark_white_random_snps <- midark_white_random_snps %>%
  mutate(snp_num = factor(snp_num), snp_id = factor(snp_id))

midark_white_200_panel_snp_names <- merge(midark_white_200_panel_snps, midark_white_random_snps, by = "snp_num")

#MIDARK and SKIA 200 diagnostic SNP names
midark_skia_random_snps <- read_excel("../../analysis/hybriddetective_analysis/snps_randomized_names/MIDARK_SKIA_random_snp_names.xlsx")

midark_skia_200_panel_snps <- read_excel("../../analysis/hybriddetective_analysis/snps_randomized_names/MIDARK_SKIA_random_200_loci_snp_names.xlsx")

midark_skia_200_panel_snps <- midark_skia_200_panel_snps %>%
  mutate(snp_num = factor(snp_num))

midark_skia_random_snps <- midark_skia_random_snps %>%
  mutate(snp_num = factor(snp_num), snp_id = factor(snp_id))

midark_skia_200_panel_snp_names <- merge(midark_skia_200_panel_snps, midark_skia_random_snps, by = "snp_num")

#Write out text files for both snp name panels
write.table(midark_white_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/snps_randomized_names/midark_white_200_panel_snp_names.txt", sep="\t")

write.table(midark_skia_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/snps_randomized_names/midark_skia_200_panel_snp_names.txt", sep="\t")
```

## Prepare 200 panel genepop files (with all individuals for each set of parents) for simulation and analysis
```{r}
#create a dataframe to designate two different populations (Pure Neosho, Pure Northern). This must be done to make sure the genepop comes out in the correct format
pop_rename_midark_white <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

pop_rename_midark_skia <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","SKIA","SKIA","SKIA","SKIA","SKIA"))


#Get full parent genepops for simulation
subset_genepop_rename("../../analysis/hybriddetective_analysis/parent_200_loci_full_genepops/MIDARK_WHITE_200_panel_genepop.txt", pop_rename_midark_white, path = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_200_loci_full_genepops/MIDARK_WHITE_edited_200_panel_genepop.txt")

subset_genepop_rename("../../analysis/hybriddetective_analysis/parent_200_loci_full_genepops/MIDARK_SKIA_200_panel_genepop.txt", pop_rename_midark_skia, path = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/analysis/hybriddetective_analysis/edited_parent_200_loci_full_genepops/MIDARK_SKIA_edited_200_panel_genepop.txt")
```

## Simulate 100 individuals in each of 6 Hybrid Categories for both parent sets (MIDARK and WHITE and MIDARK and SKIA)
```{r}
#Simulate for MIDARK and WHITE
freqbasedsim_GTFreq(GenePopData="../../analysis/hybriddetective_analysis/edited_parent_200_loci_full_genepops/MIDARK_WHITE_edited_200_panel_genepop.txt", 
                    NumSims = 1, 
                    NumReps = 1,
                    sample.sizePure = 100,
                    sample.sizeF1 = 100,
                    sample.sizeF2 = 100,
                    sample.sizeBC = 100)

#Simulate for MIDARK and SKIA
freqbasedsim_GTFreq(GenePopData="../../analysis/hybriddetective_analysis/edited_parent_200_loci_full_genepops/MIDARK_SKIA_edited_200_panel_genepop.txt", 
                    NumSims = 1, 
                    NumReps = 1,
                    sample.sizePure = 100,
                    sample.sizeF1 = 100,
                    sample.sizeF2 = 100,
                    sample.sizeBC = 100)
```

## DAPC plotting for hybridlab simulations
### In order to convert the simulated genepop data into a genind formatted file for DAPC in adegenet, I had to create a matrix of genotypes, rather than using a standard genepop format (I think there is a way to use a genepop file strictly, but I couldn't figure it out). My genotype matrix consisted of one row per individual sample, and each column was a SNP (200 total). Genotypes were coded as ###### (six numbers in a row)
```{r}
## Run DAPC for MIDARK and WHITE parents, along with admixed populations (BARON, CANEY, ILLINOIS)
MIDARK_WHITE_all_admix_genepop <- read_excel("../../analysis/hybriddetective_analysis/all_simulation_data/MIDARK_WHITE_all_hybrids.xlsx")

MIDARK_WHITE_all_admix_genepop <- as.data.frame(MIDARK_WHITE_all_admix_genepop)
## Create population vector
admix_pops_MIDARK_WHITE <- factor(c(rep("Simulated MIDARK", times = 100), rep("Simulated WHITE", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("Backcross to MIDARK", times = 100), rep("Backcross to WHITE", times = 100), rep("Empirical MIDARK", times = 16), rep("Empirical WHITE", times = 10), rep("ELK", times = 6), rep("BAYOU", times = 7), rep("UPPARK", times = 18)))

MIDARK_WHITE_all_admix_genepop <- column_to_rownames(MIDARK_WHITE_all_admix_genepop, "ind")
MIDARK_WHITE_all_admix_genepop <- as.matrix(MIDARK_WHITE_all_admix_genepop)

## Get genind file
MIDARK_WHITE_all_admix_genepop_genind <- df2genind(MIDARK_WHITE_all_admix_genepop, 
                                                ncode = 3, 
                                                pop = admix_pops_MIDARK_WHITE, 
                                                ploidy = 2)
## Scale genind file
MIDARK_WHITE_all_admix_genepop_genind <- scaleGen(MIDARK_WHITE_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

## Run DAPC
dapc_midark_white_admix <- dapc(MIDARK_WHITE_all_admix_genepop_genind, admix_pops_MIDARK_WHITE)

## PCs retained: 10
## Discriminant Functions retained: 2



## Run DAPC for MIDARK and SKIA parents, along with admixed population (ILLI)
MIDARK_SKIA_all_admix_genepop <- read_excel("../../analysis/hybriddetective_analysis/all_simulation_data/MIDARK_SKIA_all_hybrids.xlsx")

MIDARK_SKIA_all_admix_genepop <- as.data.frame(MIDARK_SKIA_all_admix_genepop)
## Create population vector
admix_pops_MIDARK_SKIA <- factor(c(rep("Simulated MIDARK", times = 100), rep("Simulated SKIA", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("Backcross to MIDARK", times = 100), rep("Backcross to SKIA", times = 100), rep("Empirical MIDARK", times = 16), rep("Empirical SKIA", times = 5), rep("ILLI", times = 9)))

MIDARK_SKIA_all_admix_genepop <- column_to_rownames(MIDARK_SKIA_all_admix_genepop, "ind")
MIDARK_SKIA_all_admix_genepop <- as.matrix(MIDARK_SKIA_all_admix_genepop)

## Get genind file
MIDARK_SKIA_all_admix_genepop_genind <- df2genind(MIDARK_SKIA_all_admix_genepop, 
                                                ncode = 3, 
                                                pop = admix_pops_MIDARK_SKIA, 
                                                ploidy = 2)
## Scale genind file
MIDARK_SKIA_all_admix_genepop_genind <- scaleGen(MIDARK_SKIA_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

## Run DAPC
dapc_midark_skia_admix <- dapc(MIDARK_SKIA_all_admix_genepop_genind, admix_pops_MIDARK_SKIA)

## PCs retained: 10
## Discriminant Functions retained: 2

```

## Make plots
```{r}
## Plot MIDARK and WHITE admixed
dapc_midark_white_admix_LDs <- as.data.frame(dapc_midark_white_admix$ind.coord)
dapc_midark_white_admix_LDs <- cbind(data.frame(admix_pops_MIDARK_WHITE), dapc_midark_white_admix_LDs)
colnames(dapc_midark_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_midark_white_admix_LDs$pop <- factor(dapc_midark_white_admix_LDs$pop, levels = c("Simulated MIDARK","Empirical MIDARK","Backcross to MIDARK", "F2", "F1", "Backcross to WHITE","Simulated WHITE","Empirical WHITE", "ELK","BAYOU","UPPARK"))

dapc_midark_white_admix_plot <- ggplot(dapc_midark_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Hybrid Category") +
   scale_fill_manual(values = c("grey5","deepskyblue","grey35","grey60","grey18","grey90","grey100","deeppink2","sienna4","lightgreen","orchid1")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.6,0.8)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 15))

options(scipen = 999)
## Plot MIDARK and SKIA admixed
dapc_midark_skia_admix_LDs <- as.data.frame(dapc_midark_skia_admix$ind.coord)
dapc_midark_skia_admix_LDs <- cbind(data.frame(admix_pops_MIDARK_SKIA), dapc_midark_skia_admix_LDs)
colnames(dapc_midark_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_midark_skia_admix_LDs$pop <- factor(dapc_midark_skia_admix_LDs$pop, levels = c("Simulated MIDARK", "Empirical MIDARK", "Backcross to MIDARK","F2","F1","Backcross to SKIA","Simulated SKIA","Empirical SKIA","ILLI"))

dapc_midark_skia_admix_plot <- ggplot(dapc_midark_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Hybrid Category") +
   scale_fill_manual(values = c("grey5","deepskyblue","grey35","grey60","grey18","grey90","grey100","navyblue","mediumpurple")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.05,0.2)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 15)) +
   theme(axis.title.y = element_blank())

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybriddetective_figures/hybrid_simulation_plots.pdf", width=16, height=7)

plot_grid(dapc_midark_white_admix_plot,
          dapc_midark_skia_admix_plot,
          nrow = 1,
          labels = c("a","b"),
          label_size = 25)

dev.off()

```

## BLACK AND WHITE plots
```{r}
## HSYC SKIA Plot
dapc_hsyc_skia_admix_LDs <- as.data.frame(dapc_hsyc_skia_admix$ind.coord)
dapc_hsyc_skia_admix_LDs <- cbind(data.frame(admix_pops), dapc_hsyc_skia_admix_LDs)
colnames(dapc_hsyc_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_skia_admix_plot_black <- ggplot(dapc_hsyc_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "grey44","grey100", "firebrick3","grey84","grey64","deepskyblue","mediumpurple","navyblue", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(legend.text = element_text(color = "white")) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_plot_black <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("lightgreen", "grey44","grey100", "sienna4", "lightyellow","lightsalmon","grey84","grey64","deepskyblue","orchid1", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20, color = "white")) +
   theme(axis.text = element_text(size = 20, color = "white")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybriddetective_figures/hyb_simulation_plots_black.pdf", width=9, height=12)

plot_grid(dapc_hsyc_white_admix_plot_black, dapc_hsyc_skia_admix_plot_black, nrow = 2, labels = c("a","b"), label_size = 27)

dev.off()
```
