---
title: "SMB_UNUSED_STUFF"
author: "Joe Gunn"
date: "3/26/2020"
output: html_document
---

```{r}
#Run first pass analysis with 30 replicates
cv_dapc <- xvalDapc(smb_noBFC10_genind, smb_phylo_pops_noBFC10, n.pca.max = 100, training.set = 0.9, result = "groupMean", xval.plot = TRUE)
#n.rep = 30 (default)

cv_dapc[2:6]


#10 PCs results in highest mean success of placing individuals in original populations
#10 PCs results in the lowest mean squared error (MSE)


cv_dapc_results <- as.data.frame(cv_dapc$`Cross-Validation Results`)
cv_dapc_results_summary <- cv_dapc_results %>% group_by(n.pca) %>% summarize(mean = mean(success), sd = sd(success))



dapc_pass1 <- ggplot(cv_dapc_results_summary, aes(x = n.pca, y = mean)) + geom_point() + geom_errorbar(aes(ymax = mean + sd, ymin = mean - sd, width = 1)) + scale_x_continuous("Number of Principal Components", labels = as.character(cv_dapc_results_summary$n.pca), breaks = cv_dapc_results_summary$n.pca) + theme_set(theme_cowplot(12)) + labs(y = "Proportion Classified Successfully") + theme(axis.title = element_text(size = 15)) + theme(axis.text = element_text(size = 15)) + theme(axis.title.x = element_blank())



###############

#Run second pass analysis between pcs 2 and 30 (the straight linein the abvoe plot, with 1000 reps per number of PCs)
cv_dapc_narrow <- xvalDapc(smb_noBFC10_genind, smb_phylo_pops_noBFC10, n.pca = 2:30, n.rep = 1000, result = "groupMean", xval.plot = TRUE)

cv_dapc_narrow[2:6]

cv_narrow_results <- as.data.frame(cv_dapc_narrow$`Cross-Validation Results`)

#Best number of pcs = 9 pcs, 82.27 % success
#Root mean squared error lowest = 6 pcs, 0.1951

cv_narrow_results_other <- cv_narrow_results %>% group_by(n.pca) %>% summarize(mean = mean(success), sd = sd(success))

dapc_pass2 <- ggplot(cv_narrow_results_other, aes(x = n.pca, y = mean)) + geom_point() + geom_errorbar(aes(ymax = mean + sd, ymin = mean - sd, width = 1)) + scale_x_continuous("Number of Principal Components", labels = as.character(cv_narrow_results_other$n.pca), breaks = cv_narrow_results_other$n.pca) + theme_set(theme_cowplot(12)) + theme(axis.title = element_text(size = 15)) + theme(axis.text = element_text(size = 15)) + labs(y = "Proportion Classified Successfully")


#Both Plots together
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Smallmouth_Bass_Genomics/SMB_GENOMICS_REVAMPED/dapc_cv_plots.pdf", width=10, height=12)

plot_grid(dapc_pass1, dapc_pass2, nrow = 2, labels = c("a", "b"), label_size = 20, label_x = 0, label_y = 1.02)

dev.off()


```


```{r}

#Run DAPC
all_clusters <- find.clusters(all_noBCF10_genind, max.n.clust = 40)
all_dapc1 <- dapc(all_noBCF10_genind, all_clusters$grp)

#8 clusters
               
all_dapc1$posterior
contrib <- loadingplot(all_dapc1$var.contr, axis = 1, thres = 0.0003, lab.jitter = 1)
all_dapc1$var.contr
## INFO #########################
#Number of PCs retained: 6
#Number of discriminant functions used: 3
#################################

all_clusters$grp


#Plot DAPC
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Smallmouth_Bass_Genomics/SMB_GENOMICS_REVAMPED/DAPC_all_plot.pdf", width=8, height=5) 

scatter(dapc_all_noBFC10_bypop, posi.da = "bottomleft", scree.pca = T, posi.pca = "topleft", bg = "white", ratio.pca = 0.3, cex = 3, cell = 0, col = c("navyblue","forestgreen","chocolate1","deeppink2","darkviolet","goldenrod3", "cyan", "seagreen2"))

dev.off()

```

#Run DAPC for just smallmouth bass samples

```{r}
smb_clusters <- find.clusters(smb_genind, max.n.clust = 40)

smb_dapc1 <- dapc(smb_genind, smb_clusters$grp)


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Projects/Smallmouth_Bass_Genomics/SMB_GENOMICS_REVAMPED/DAPC_all_plot.pdf", width=8, height=5) 

scatter(smb_dapc1, posi.da = "topleft", scree.pca = T, posi.pca = "bottomleft", bg = "white", ratio.pca = 0.3, cex = 3, cell = 0, col = c("forestgreen","navyblue","darkviolet","cyan","deeppink2","goldenrod3"))

dev.off()

#get data for K=7
smb_group_members_k6 <- as.data.frame(smb_clusters$grp)
smb_clusters$grp

bic_stats_k7 <- as.data.frame(smb_clusters$Kstat)

#Write out group membershp data to useable table
write.table(group_members_k7,"~/Desktop/Grad_School_Stuff/Projects/Smallmouth_Bass_Genomics/SMB_GENOMICS_REVAMPED/DAPC_group_membership_K7.txt", sep="\t")


```

``{r setup, include=FALSE}
library(radiator)
library(readxl)
library(tidyverse)
library(cowplot)
library(devtools)
library(SNPRelate)
library(vcfR)
library(pophelper)
library(PopGenome)
library(pcadapt)
source("./vcf2sfs.r")
```
##unused hybrid detective stuff


#Format genepop file for simulated HSYC and WHITE parent 1 and 2 
#pop_rename_hsyc_white_p1_p2 <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_P1_P2_sim.txt", pop_rename_hsyc_white_p1_p2, path = "/Users/joegunn/Desktop/hsyc_white_P1_P2_sim_2pop.txt")

#Format genepop file for simulated HSYC and WHITE F1 backcross to HSYC 
#pop_rename_hsyc_white_F1_HSYC <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_F1_HSYC_sim.txt", pop_rename_hsyc_white_F1_HSYC, path = "/Users/joegunn/Desktop/hsyc_white_F1_HSYC_sim_2pop.txt")

#Format genepop file for simulated HSYC and WHITE F1 backcross to WHITE 
#pop_rename_hsyc_white_F1_WHITE <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_white_F1_WHITE_sim.txt", pop_rename_hsyc_white_F1_WHITE, path = "/Users/joegunn/Desktop/hsyc_white_F1_WHITE_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA parent 1 and 2 
#pop_rename_hsyc_skia_p1_p2 <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P1","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2","P2"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_P1_P2_sim.txt", pop_rename_hsyc_skia_p1_p2, path = "/Users/joegunn/Desktop/hsyc_skia_P1_P2_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA F1 backcross to HSYC 
#pop_rename_hsyc_skia_F1_HSYC <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_F1_HSYC_sim.txt", pop_rename_hsyc_skia_F1_HSYC, path = "/Users/joegunn/Desktop/hsyc_skia_F1_HSYC_sim_2pop.txt")

#Format genepop file for simulated HSYC and SKIA F1 backcross to SKIA 
#pop_rename_hsyc_skia_F1_SKIA <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","SKIA","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1","F1"))

#subset_genepop_rename("../../raw_data/genepop_data/hybrid_lab_simulations/text_files/hsyc_skia_F1_SKIA_sim.txt", pop_rename_hsyc_skia_F1_SKIA, path = "/Users/joegunn/Desktop/hsyc_skia_F1_SKIA_sim_2pop.txt")
```

## Format all hybrid datafiles
```{r}
##WHITE RIVER ADMIXED POPS
#pop_rename_white_bayou <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("AT","AT","AT","AT","AT","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BAYOU_genepop.txt", pop_rename_white_bayou, path = "/Users/joegunn/Desktop/hsyc_white_BAYOU_genepop_correct.txt")


#pop_rename_white_bsug <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("MI","MI","MI","MI","MI","MI","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BSUG_genepop.txt", pop_rename_white_bsug, path = "/Users/joegunn/Desktop/hsyc_white_BSUG_genepop_correct.txt")


#pop_rename_white_buff <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("BC","BC","BC","BC","BC","BC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_BUFF_genepop.txt", pop_rename_white_buff, path = "/Users/joegunn/Desktop/hsyc_white_BUFF_genepop_correct.txt")


#pop_rename_white_elk <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("ELK","ELK","ELK","ELK","ELK","ELK","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_ELK_genepop.txt", pop_rename_white_elk, path = "/Users/joegunn/Desktop/hsyc_white_ELK_genepop_correct.txt")


#pop_rename_white_spav <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("SPVW","SPVW","SPVW","SPVW","SPVW","SPVW","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_white_admixed_vcf/hsyc_white_SPAV_genepop.txt", pop_rename_white_spav, path = "/Users/joegunn/Desktop/hsyc_white_SPAV_genepop_correct.txt")



#SKIATOOK LAKE ADMIXED POPS
#pop_rename_skia_baron <- data.frame(Opop = c("","","","","","","","","","","","","","","","","",""), Rename = c("BFORK","BFORK","BFORK","BFORK","BFORK","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_BARON_genepop.txt", pop_rename_skia_baron, path = "/Users/joegunn/Desktop/hsyc_skia_BARON_genepop_correct.txt")


#pop_rename_skia_caney <- data.frame(Opop = c("","","","","","","","","","","","","","","","","",""), Rename = c("CANEY","CANEY","CANEY","CANEY","CANEY","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_CANEY_genepop.txt", pop_rename_skia_caney, path = "/Users/joegunn/Desktop/hsyc_skia_CANEY_genepop_correct.txt")


#pop_rename_skia_illinois <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","",""), Rename = c("ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","ILLINOIS","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","HSYC","SKIA","SKIA","SKIA","SKIA","SKIA"))

#subset_genepop_rename("../../raw_data/genepop_data/hsyc_skia_admixed_vcf/hsyc_skia_ILLINOIS_genepop.txt", pop_rename_skia_illinois, path = "/Users/joegunn/Desktop/hsyc_skia_ILLINOIS_genepop_correct.txt")


#genepop_flatten("../../../../../../hsyc_skia_ILLINOIS_genepop_correct.txt")
#genepop_detective("../../../../../../hsyc_skia_ILLINOIS_genepop_correct.txt", variable = "PopNum")
```




```{r}


# Site Frequency Spectra 

##The site frequency spectrum (SFS) is calculated for each population separately. The SFS is a vector of integer values, with each integer having a corresponding x and y coordinate. the x coordinate is the derived allele frequency - first number in the list corresponds to an allele frequency of 0%, and the last number in the list corresponds to an allele frequency of 1%. The y coordinate, or the number value itself, is the number of SNP loci that have the corresponding allele frequency. 

```{r}
#All samples

#SFS pop 1

vcf2dadi("./vcf_files/vcf_filtered.vcf","bayescan_all_species.txt","pop1.fs", 1)

vcf2dadi("./vcf_files/vcf_filtered_neosho.vcf", "smb_neosho_pops.txt","pop1.fs", 1, n.digit = 4)
vcf2dadi("./vcf_files/vcf_filtered_neosho.vcf", "smb_neosho_pops.txt","pop2.fs", 2, n.digit = 4)
vcf2dadi("./vcf_files/vcf_filtered_neosho.vcf", "smb_neosho_pops.txt","pop3.fs", 3, n.digit = 4)
vcf2dadi("./vcf_files/vcf_filtered_neosho.vcf", "smb_neosho_pops.txt","pop4.fs", 4, n.digit = 4)

```


## Create hierarchical structure plots for all samples, including the hybrid discovered above (BFC10)
```{r}
# #Here I am running hierarchical structure analyses for Smallmouth Bass only (SMB), Neosho Smallmouth Bass only (NEO), and Northern Smallmouth Bass only (NOR), INCLUDING the hybrid detected above (BCF10).I identified a single potential SMB X SPB (spotted bass) hybrid based on the structure results above. I would like to know if including the hybrid in analysis of population structure - using ADMIXTURE - will change results for the best supported number of clusters. So, I am building ADMIXTURE diagrams for each hierarchical level: all samples (completed above), Smallmouth bass only, Neosho Smallmouth Bass only, Northern Smallmouth Bass only.

####Read in ADMIXTURE Q result files

admix_with_hybrid_smb <- readQ("../ADMIXTURE_analysis/admix_all_samples/pass2_smb/pass2_all_smb.5.Q") # K=5 was the best supported value of K with CV error = 0.288

admix_with_hybrid_neosho <- readQ("../ADMIXTURE_analysis/admix_all_samples/pass2_neosho/pass2_all_neosho.1.Q") # K=1 was the best supported value of K with CV error = 0.294

admix_with_hybrid_northern <- readQ("../ADMIXTURE_analysis/admix_all_samples/pass2_northern/pass2_all_northern.1.Q") # K=1 was the best supported value of K with CV error = 0.306

###SMALLMOUTH BASS ONLY###
plotQ(admix_with_hybrid_smb,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_hybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03","ER35","ER20","SPRMO11","ER33","ER17","ER30","ER44","MI419","GLVR4","SPRMO19","NOIS18","BC01","BC07","BC14","MI425","SPVW12","SPVW07","SPVW05","BC08","BC12","SPVW11","BC09","MI420","MI423","MI422","MI421","SPVW02","SPVW14","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK30","BFORK49","CANEY15","GRSPB02","GRSPB03","NOIS07","NOIS12","NOIS08","BFORK24","BFC10","AR30","AR29","AT02","AT09","AT12","AT08","AT05","BFC02","BFC06","BFC49","AR18","AR19","AR21","AR16"),
      clustercol = c("deepskyblue", "deeppink2", "mediumpurple", "forestgreen","chocolate1"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.8, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admixture_withhybrid_smb_k5",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

###NEOSHO ONLY###
plotQ(admix_with_hybrid_neosho,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_hybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("ER35","ER20","SPRMO11","ER33","ER17","ER30","ER44","MI419","GLVR4","SPRMO19","NOIS18","BC01","BC07","BC14","MI425","SPVW12","SPVW07","SPVW05","BC08","BC12","SPVW11","BC09","MI420","MI423","MI422","MI421","SPVW02","SPVW14","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK30","BFORK49","CANEY15","GRSPB02","GRSPB03","NOIS07","NOIS12","NOIS08","BFORK24","BFC10","AR30","AR29","AT02","AT09","AT12","AT08","AT05","BFC02","BFC06","BFC49","AR18","AR19","AR21","AR16"),
      clustercol = c("deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admixture_withhybrid_neosho_k1",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

###NORTHERN ONLY###
plotQ(admix_with_hybrid_northern,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_hybrid_northern,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03"),
      clustercol = c("deeppink2"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 2, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_withhybrid_northern_k1",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)
```

## Create hierarchical structure plots for all non-hybrid samples, excluding the hybrid discovered above (BFC10)
```{r}
#Here I am running hierarchical structure analyses for all samples (ALL), Smallmouth Bass only (SMB), Neosho Smallmouth Bass only (NEO), and Northern Smallmouth Bass only (NOR), EXCLUDING the hybrid detected above (BCF10). These results will be used to show if there is a difference in structure inferred when hybrids between species are included or not included in the analysis.

####Read in ADMIXTURE Q result files

admix_nohybrid_all <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_all/pass2_noBFC10.5.Q") # K=5 was the best supported value of K with CV error = 0.25002 (vs. K = 4, which had a best K value of 0.2501)

admix_nohybrid_smb <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.3.Q") # K=3 was the best supported value of K with CV error = 0.254

admix_nohybrid_neosho <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.3.Q") # K=3 was the best supported value of K with CV error = 0.256

admix_nohybrid_northern <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_northern/pass2_noBFC10_northern.1.Q") # K=1 was the best supported value of K with CV error = 0.306


###ALL SAMPLES###
plotQ(admix_nohybrid_all,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_all,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB51","GRSPB52","GLVR11","SPRMO49","GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03","ER35","ER20","SPRMO11","ER33","ER30","ER17","ER44","GLVR4","MI419","BC07","BC14","MI425","SPVW12","SPVW07","SPVW02","SPVW05","BC08","BC12","SPVW11","BC09","MI420","MI423","MI422","MI421","BC01","SPVW14","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK30","BFORK49","CANEY15","GRSPB02","GRSPB03","NOIS07","NOIS12","NOIS08","BFORK24","NOIS18","SPRMO19","AR30","AR29","AT02","AT09","AT12","AT08","AT05","BFC02","BFC06","BFC49","AR19","AR18","AR21","AR16"),
      clustercol = c("forestgreen", "goldenrod3", "navyblue", "deepskyblue", "deeppink2"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_all_k5",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

###SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03","ER35","ER20","SPRMO11","ER33","ER30","ER17","ER44","GLVR4","MI419","BC07","BC14","MI425","SPVW12","SPVW07","SPVW02","SPVW05","BC08","BC12","SPVW11","BC09","MI420","MI423","MI422","MI421","BC01","SPVW14","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK30","BFORK49","CANEY15","GRSPB02","GRSPB03","NOIS07","NOIS12","NOIS08","BFORK24","NOIS18","SPRMO19","AR30","AR29","AT02","AT09","AT12","AT08","AT05","BFC02","BFC06","BFC49","AR19","AR18","AR21","AR16"),
      clustercol = c("deeppink2", "forestgreen", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k3",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("ER35","ER20","SPRMO11","ER33","ER30","ER17","ER44","GLVR4","MI419","BC07","BC14","MI425","SPVW12","SPVW07","SPVW02","SPVW05","BC08","BC12","SPVW11","BC09","MI420","MI423","MI422","MI421","BC01","SPVW14","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK30","BFORK49","CANEY15","GRSPB02","GRSPB03","NOIS07","NOIS12","NOIS08","BFORK24","NOIS18","SPRMO19","AR30","AR29","AT02","AT09","AT12","AT08","AT05","BFC02","BFC06","BFC49","AR19","AR18","AR21","AR16"),
      clustercol = c("deeppink2", "forestgreen", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k3",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

###NORTHERN SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_northern,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_northern,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03"),
      clustercol = c("deeppink2"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 2, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_northern_k1",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)
```

### Same analyses and data, but in the order of the FINERADSTRUCTURE results 
### Neosho Only
```{r}
admix_nohybrid_neosho_k3 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.3.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho_k3,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29"),
      clustercol = c("deeppink2", "forestgreen", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k3_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_neosho_k4 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.4.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho_k4,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29"),
      clustercol = c("forestgreen", "deeppink2", "mediumpurple","deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k4_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_neosho_k5 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.5.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho_k5,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29"),
      clustercol = c("deepskyblue", "forestgreen", "mediumpurple","orchid1", "deeppink2"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k5_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_neosho_k6 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.6.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho_k6,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29"),
      clustercol = c("forestgreen", "lightgreen", "mediumpurple","deeppink2", "deepskyblue", "orchid1"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k6_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_neosho_k7 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_neosho/pass2_noBFC10_neosho.7.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_neosho_k7,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_neosho,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29"),
      clustercol = c("deepskyblue", "lightsalmon", "mediumpurple","orchid1", "lightgreen", "deeppink2", "forestgreen"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_neosho_k7_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)
```

### Same analyses and data, but in the order of the FINERADSTRUCTURE results 
### All SMB
```{r}

admix_nohybrid_smb_k2 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.2.Q") # K=2 was the best supported value of K with CV error = 0.254

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k2,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("deepskyblue","deeppink2"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k2_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k3 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.3.Q") # K=3 was the best supported value of K with CV error = 0.254

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k3,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("deeppink2", "forestgreen", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k3_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k4 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.4.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k4,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("forestgreen", "deeppink2", "mediumpurple","deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k4_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k5 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.5.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k5,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("mediumpurple", "forestgreen", "deeppink2","deepskyblue", "navyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k5_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k6 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.6.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k6,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("mediumpurple", "deepskyblue", "navyblue","chocolate1", "deeppink2", "forestgreen"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k6_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k7 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.7.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k7,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("chocolate1", "lightgreen", "mediumpurple","navyblue", "forestgreen", "deeppink2", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k7_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k8 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.8.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k8,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("orchid1", "mediumpurple", "deepskyblue","chocolate1", "deeppink2", "lightgreen", "forestgreen","navyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k8_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k9 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.9.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k9,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("mediumpurple", "deepskyblue", "chocolate1","lightgreen", "orchid1", "navyblue", "deeppink2","forestgreen","lightyellow"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k9_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

admix_nohybrid_smb_k10 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/pass2_noBFC10_smb/pass2_noBFC10_smb.10.Q") # K=3 was the best supported value of K with CV error = 0.256

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_nohybrid_smb_k10,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("deepskyblue", "lightyellow", "navyblue","mediumpurple", "deeppink2", "sienna4", "lightgreen","chocolate1","forestgreen", "orchid1"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_NOhybrid_smb_k10_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)
```

### All SMB, 1000 highest FSTs
```{r}

admix_1000_fst_smb_k2 <- readQ("../ADMIXTURE_analysis/admix_all_samples_noBFC10/admix_results_smb_1000_fst/smb_1000_fst.2.Q") # K=3 was the best supported value of K with CV error = 0.254

###NEOSHO SMALLMOUTH BASS ONLY###
plotQ(admix_1000_fst_smb_k2,
      showindlab = T, 
      useindlab=F,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_nohybrid_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("SPRMO19","CANEY15","SPRMO11","GLVR4","SPVW14","SPVW05","SPVW02","MI421","MI422","SPVW07","SPVW11","BC01","BC09","BC08","BC12","BC07","MI420","MI423","MI419","MI425","BC14","SPVW12","ER35","ER33","ER20","ER17","ER30","ER44","BFORK49","BFORK24","BFORK30","NOIS07","NOIS12","GRSPB03","GRSPB02","BFORK23","NOIS18","NOIS08","GRSPB69","CANEY16","BFORK02","HC44","HC42","HC41","HC38","SC06","SC37","HC43","HC40","BFC02","BFC49","BFC06","AR16","AR19","AR21","AR18","AT05","AT02","AT09","AT08","AT12","AR30","AR29","STOCK05","STOCK06","STOCK03","STOCK04","BP10","BP01","BP17","BP02","BP07","SKIA06","SKIA05","SKIA04","SKIA09","SKIA03","TBLR02","TBLR04","TBLR01","TBLR03","GRSPB50","GRSPB39","GRSPB41","GRSPB35","GRSPB36","GRSPB37"),
      clustercol = c("deeppink2", "deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = F, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/admix_1000_fst_smb_k2_finerad_order",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)

```

## Read in three datafiles, each representing a unique subset of populations within the data.
```{r}
#Each dataset contains 50,828 rows, one for each individual snp along with the corresponding information

#Read in tables of Fst values from BayeScan
#All samples
bayes_all_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/from_finerad/smb_all_bayes_noBFC10_results_taxon_fst.xlsx")

#just smallmouth bass samples
bayes_smb_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/not_from_finerad/smb_smb_bayes_noBFC10_results_fst.xlsx") 
bayes_smb_finerad_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/from_finerad/smb_smb_bayes_noBFC10_finerad_results_fst.xlsx") 

#just neosho samples
bayes_neosho_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/not_from_finerad/neo_bayes_noBFC10_results_fst.xlsx") 
bayes_neosho_finerad_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/from_finerad/neo_bayes_noBFC10_finerad_results_fst.xlsx") 

#just northern samples
bayes_northern_fst <- read_excel("../../raw_data/outlier_fst_data/bayescan_data/from_finerad/nor_bayes_results_fst.xlsx") 
```

## Data Filtering

###The data filtering section identifies which individual SNPs from the Bayescan program output are considered significant outliers. According to the Bayescan user manual, SNPs with a log of posterior odds value of 1.5 or greater are considered substantially likely to be under selection. This section therefore identifies individual SNPs in the dataset that have a logPO value greater than or equal to 1.5 and labels them as "signficant". Alternatively, all SNPs with a logPO value less than 1.5 are identified as "not significant". A new data file with additional metadata identifying significant SNPs is then created for plotting

## Get Significant and Insignificant SNPs for ALL data Only
```{r}
#Rearrange data files so that significant SNPs (or log10PO is 1.5 or greater) are labeled
 
bayescan_all_fst_significant <- bayes_all_fst %>% 
      filter(logPO >= 0.5) #extract only SNPs with logPO greater than or equal to 1.5 (significant)

bayescan_all_fst_not_significant <- bayes_all_fst %>% 
      filter(logPO < 0.5) #extract only SNPs with logPO less than 1.5 (not significant)

dim(bayescan_all_fst_significant) #3 total significant SNP
dim(bayescan_all_fst_not_significant) #50,825 insignificant SNPs

#create separate columns of metadata for both datasets created in the above step (significant and not significant), which will be later appended to each dataset
sig_label_all <- as.data.frame(c(rep("significant", times = 3))) #no significant snps
      colnames(sig_label_all) <- c("selection")
      
non_sig_label_all <- as.data.frame(c(rep("insignificant", times = 50825))) #all non-significant snps
      colnames(non_sig_label_all) <- c("selection")
      
sig_level_label_all <- as.data.frame(c(rep("very_strong", times = 1), rep("substantial", times = 2), rep("not_significant", times = 50825)))
      colnames(sig_level_label_all) <- c("sig_level")


#append metadata to each respective dataset
bayescan_all_fst_significant <- cbind(bayescan_all_fst_significant, sig_label_all)
bayescan_all_fst_not_significant <- cbind(bayescan_all_fst_not_significant, non_sig_label_all)

#combine both significant and non-significant datasets into one dataset with all data, now appropriately labeled
bayescan_all_fst_almost_ready <- rbind(bayescan_all_fst_significant, bayescan_all_fst_not_significant)
bayescan_all_fst_ready <- cbind(bayescan_all_fst_almost_ready, sig_level_label_all)
```

## Get Significant and Insignificant SNPs for SMB data Only
```{r}
#Rearrange data files so that significant SNPs (or log10PO is 1.5 or greater) are labeled
 
bayescan_smb_finerad_fst_significant <- bayes_smb_finerad_fst %>% 
      filter(logPO >= 0.5) #extract only SNPs with logPO greater than or equal to 1.5 (significant)

bayescan_smb_finerad_fst_not_significant <- bayes_smb_finerad_fst %>% 
      filter(logPO < 0.5) #extract only SNPs with logPO less than 1.5 (not significant)

dim(bayescan_smb_finerad_fst_significant) #972 total significant SNPs
dim(bayescan_smb_finerad_fst_not_significant) #49,856 insignificant SNPs

#create separate columns of metadata for both datasets created in the above step (significant and not significant), which will be later appended to each dataset
sig_label_smb_finerad = as.data.frame(c(rep("significant", times = 972))) #no significant snps
      colnames(sig_label_smb_finerad) <- c("selection")
      
non_sig_label_smb_finerad <- as.data.frame(c(rep("insignificant", times = 49856))) #all non-significant snps
      colnames(non_sig_label_smb_finerad) <- c("selection")

sig_level_label_smb <- as.data.frame(c(rep("decisive", times = 176), rep("very_strong", times = 135), rep("strong", times = 246), rep("substantial", times = 415), rep("not_significant", times = 49856)))
      colnames(sig_level_label_smb) <- c("sig_level")      
      
#append metadata to each respective dataset
bayescan_smb_finerad_fst_significant <- cbind(bayescan_smb_finerad_fst_significant, sig_label_smb_finerad)
bayescan_smb_finerad_fst_not_significant <- cbind(bayescan_smb_finerad_fst_not_significant, non_sig_label_smb_finerad)

#combine both significant and non-significant datasets into one dataset with all data, now appropriately labeled
bayescan_smb_finerad_fst_almost_ready <- rbind(bayescan_smb_finerad_fst_significant, bayescan_smb_finerad_fst_not_significant)
bayescan_smb_finerad_fst_ready <- cbind(bayescan_smb_finerad_fst_almost_ready, sig_level_label_smb)
```

## Get Significant and Insignificant SNPs for Neosho data only
```{r}

bayescan_neosho_finerad_fst_significant <- bayes_neosho_finerad_fst %>% 
      filter(logPO >= 0.5) #extract only SNPs with logPO greater than or equal to 1.5 (significant)

bayescan_neosho_finerad_fst_not_significant <- bayes_neosho_finerad_fst %>% 
      filter(logPO < 0.5) #extract only SNPs with logPO less than 1.5 (not significant)

dim(bayescan_neosho_finerad_fst_significant) #40 total significant SNPs
dim(bayescan_neosho_finerad_fst_not_significant) #50,788 total insignificant SNPs

#create separate columns of metadata for both datasets created in the above step (significant and not significant), which will be later appended to each dataset
sig_label_neosho_finerad = as.data.frame(c(rep("significant", times = 40)))
      colnames(sig_label_neosho_finerad) <- c("selection")
      
non_sig_label_neosho_finerad <- as.data.frame(c(rep("insignificant", times = 50788)))
      colnames(non_sig_label_neosho_finerad) <- c("selection")
      
sig_level_label_neosho <- as.data.frame(c(rep("decisive", times = 3), rep("very_strong", times = 10), rep("strong", times = 15), rep("substantial", times = 12), rep("not_significant", times = 50788)))
      colnames(sig_level_label_neosho) <- c("sig_level")
      
#append metadata to each respective dataset
bayescan_neosho_finerad_fst_significant <- cbind(bayescan_neosho_finerad_fst_significant, sig_label_neosho_finerad)
bayescan_neosho_finerad_fst_not_significant <- cbind(bayescan_neosho_finerad_fst_not_significant, non_sig_label_neosho_finerad)

#combine both significant and non-significant datasets into one dataset with all data, now appropriately labeled
bayescan_neosho_finerad_fst_almost_ready <- rbind(bayescan_neosho_finerad_fst_significant, bayescan_neosho_finerad_fst_not_significant)
bayescan_neosho_finerad_fst_ready <- cbind(bayescan_neosho_finerad_fst_almost_ready, sig_level_label_neosho)
```

## Get Significant and Insignificant SNPs for Northern data only
```{r}
bayescan_northern_fst_significant <- bayes_northern_fst %>% 
      filter(logPO >= 0.5) #extract only SNPs with logPO greater than or equal to 1.5 (significant)

bayescan_northern_fst_not_significant <- bayes_northern_fst %>% 
      filter(logPO < 0.5) #extract only SNPs with logPO less than 1.5 (not significant)


dim(bayescan_northern_fst_significant) # 6 significant snps
dim(bayescan_northern_fst_not_significant) #50,822 neutral snps


#create separate columns of metadata for both datasets created in the above step (significant and not significant), which will be later appended to each dataset
sig_label_northern = as.data.frame(c(rep("significant", times = 6))) #no significant snps
      colnames(sig_label_northern) <- c("selection")
non_sig_label_northern <- as.data.frame(c(rep("insignificant", times = 50822))) #all non-significant snps
      colnames(non_sig_label_northern) <- c("selection")
      
sig_level_label_northern <- as.data.frame(c(rep("substantial", times = 6), rep("not_significant", times = 50822)))
      colnames(sig_level_label_northern) <- c("sig_level")

#append metadata to each respective dataset
bayescan_northern_fst_significant <- cbind(bayescan_northern_fst_significant, sig_label_northern)
bayescan_northern_fst_not_significant <- cbind(bayescan_northern_fst_not_significant, non_sig_label_northern)

#combine both significant and non-significant datasets into one dataset with all data, now appropriately labeled
bayescan_northern_fst_almost_ready <- rbind(bayescan_northern_fst_significant, bayescan_northern_fst_not_significant)
bayescan_northern_fst_ready <- cbind(bayescan_northern_fst_almost_ready, sig_level_label_northern)
```

## Filter datasets to omit SNPs with PO values of 1000 (because they distort the figure. Add the number of SNPs at 1000 to the final count of loci under selection)
```{r}

###smallmouth bass samples
bayescan_all_fst_1000s <- bayescan_all_fst_ready %>% 
      filter(logPO >= 1000) #extract only snps with logPO greater than or equal to 1000
bayescan_all_fst_not1000 <- bayescan_all_fst_ready %>% 
      filter(logPO < 1000) #extract only snps with log10 less than 1000

###smallmouth bass samples
bayescan_smb_fst_1000s <- bayescan_smb_finerad_fst_ready %>% 
      filter(logPO >= 1000) #extract only snps with logPO greater than or equal to 1000, #6 SNPs were over 1000
bayescan_smb_fst_not1000 <- bayescan_smb_finerad_fst_ready %>% 
      filter(logPO < 1000) #extract only snps with log10 less than 1000

###neosho samples
bayescan_neosho_fst_1000s <- bayescan_neosho_finerad_fst_ready %>% 
      filter(logPO >= 1000) #extract only snps with logPO greater than or equal to 1000, #only one is 1000, so 44 SNPs out of 45 will be in the figure
bayescan_neosho_fst_not1000 <- bayescan_neosho_finerad_fst_ready %>% 
      filter(logPO < 1000) #extract only snps with log10 less than 1000

###northern samples
bayescan_northern_fst_1000s <- bayescan_northern_fst_ready %>% 
      filter(logPO >= 1000) #extract only snps with logPO greater than or equal to 1000
bayescan_northern_fst_not1000 <- bayescan_northern_fst_ready %>% 
      filter(logPO < 1000) #extract only snps with log10 less than 1000

```

## Plot Results

The Plot Results section uses the full dataset produced in the Data Filtering section to plot Bayescan results for visualization

```{r}
#plot outliers separately - each plot will show logPO on the x axis and fst on the y axis, with significance (selection) shown in different colors

#Plot All samples
all_bayes_plot <- ggplot(bayescan_all_fst_not1000, aes(x = logPO, y = fst, fill = sig_level)) + #make plot from non-1000 logPO for all SMB 
      geom_point(aes(fill = sig_level), alpha = 0.8, color = "black", pch = 21, size = 4, show.legend = F) + #do not show legend on the plot itself
      geom_vline(xintercept = 2, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 1.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
       geom_vline(xintercept = 1, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 0.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      theme_set(theme_cowplot(12)) + #set ggplot to cowplot style 12 (gets rid of grid lines)
      scale_fill_manual(values = c("grey","#FCFFA4FF","#BB3754FF")) + #set the color of points to black
      labs(x = "Log10(Posterior Odds)", y = "FST") + #re-name the labels for the x and y axis
      theme(axis.title = element_text(size = 15, color = "black")) + #make the axis titles size 15
      theme(axis.text = element_text(size = 15, color = "black")) + #make the axis text size 15
      theme(axis.line = element_line(color = "black")) +
      theme(axis.ticks = element_line(color = "black")) +
      theme(axis.title.x = element_blank())

#Plot smallmouth bass outliers
smb_bayes_plot <- ggplot(bayescan_smb_fst_not1000, aes(x = logPO, y = fst, fill = sig_level)) + #make plot from non-1000 logPO for all SMB 
      geom_point(aes(fill = sig_level), alpha = 0.8, color = "black", pch = 21, size = 4, show.legend = F) + #do not show legend on the plot itself
      geom_vline(xintercept = 2, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 1.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
       geom_vline(xintercept = 1, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 0.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      theme_set(theme_cowplot(12)) + #set ggplot to cowplot style 12 (gets rid of grid lines)
      scale_fill_manual(values = c("#56106EFF","grey","#F98C0AFF","#FCFFA4FF","#BB3754FF")) + #set the color of points to black
      labs(x = "Log10(Posterior Odds)", y = "FST") + #re-name the labels for the x and y axis
      theme(axis.title = element_text(size = 15, color = "black")) + #make the axis titles size 15
      theme(axis.text = element_text(size = 15, color = "black")) + #make the axis text size 15
      theme(axis.line = element_line(color = "black")) +
      theme(axis.ticks = element_line(color = "black")) +
      theme(axis.title.x = element_blank()) +
      theme(axis.title.y = element_blank())

#Plot neosho outliers
neosho_bayes_plot <- ggplot(bayescan_neosho_finerad_fst_ready, aes(x = logPO, y = fst, fill = sig_level)) + #make plot from non-1000 logPO for neosho only
      geom_point(aes(fill = sig_level), alpha = 0.8, color = "black", pch = 21, size = 4, show.legend = F) + #do not show legend on the plot itself
      geom_vline(xintercept = 2, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 1.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
       geom_vline(xintercept = 1, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 0.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      theme_set(theme_cowplot(12)) + #set ggplot to cowplot style 12 (gets rid of grid lines)
      scale_fill_manual(values = c("#56106EFF","grey","#F98C0AFF","#FCFFA4FF","#BB3754FF")) + #set the color of non-significant points to black and significant points to red
      labs(x = "Log10(Posterior Odds)", y = "FST") + #re-name the labels for the x and y axis
      theme(axis.title = element_text(size = 15, color = "black")) + #make the axis titles size 15
      theme(axis.text = element_text(size = 15, color = "black")) + #make the axis text size 15
      theme(axis.line = element_line(color = "black")) +
      theme(axis.ticks = element_line(color = "black")) 

#Plot northern outliers
northern_bayes_plot <- ggplot(bayescan_northern_fst_not1000, aes(x = logPO, y = fst, fill = sig_level)) + 
      geom_point(aes(fill = sig_level), alpha = 0.8, color = "black", pch = 21, size = 4, show.legend = F) + #do not show legend on the plot itself
      geom_vline(xintercept = 2, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 1.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
       geom_vline(xintercept = 1, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      geom_vline(xintercept = 0.5, linetype = "longdash", color = "black") + #add a vertical line at the x value 1.5, and make it a dashed line
      theme_set(theme_cowplot(12)) + #set ggplot to cowplot style 12 (gets rid of grid lines)
      scale_fill_manual(values = c("grey","#FCFFA4FF")) + #set the color of points to black
      labs(x = "Log10(Posterior Odds)", y = "FST") + #re-name the labels for the x and y axis
      theme(axis.title = element_text(size = 15, color = "black")) + #make the axis titles size 15
      theme(axis.text = element_text(size = 15, color = "black")) + #make the axis text size 15
      theme(axis.title.y = element_blank()) + #get rid of the y axis label
      theme(axis.line = element_line(color = "black")) +
      theme(axis.ticks = element_line(color = "black"))
   
#Put Bayescan plots together
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/Bayescan/bayescan_plots.pdf", width=13, height=10)

plot_grid(all_bayes_plot, smb_bayes_plot, neosho_bayes_plot, northern_bayes_plot, nrow = 2, labels = c("a","b","c","d"), label_size = 25, align = "hv")

dev.off()
```


## Plot DAPC in BLACK and WHITE
```{r}
####DAPC Plots###

###SMB Samples Only###
###Outlier SNPs###
smb_outlier_dapc_plot_black <- ggplot(dapc_smb_outlier_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = F) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "lightgreen", "sienna4","lightyellow", "firebrick3", "lightsalmon", "deepskyblue", "mediumpurple","forestgreen","chocolate1","navyblue","orchid1","deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) +
   theme(axis.text = element_text(color = "white")) +
   theme(axis.title = element_text(color = "white")) 
   
   

###Neutral SNPs###
smb_neutral_dapc_plot_black <- ggplot(dapc_smb_neutral_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "lightgreen", "sienna4","lightyellow", "firebrick3", "lightsalmon", "deepskyblue", "mediumpurple","forestgreen","chocolate1","navyblue","orchid1","deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.15,0.71)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title.y = element_blank()) +
   theme(legend.text = element_text(size = 20)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) + 
   theme(plot.background = element_rect(fill = "black")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) +
   theme(axis.text = element_text(color = "white")) +
   theme(axis.title = element_text(color = "white")) 


###Neosho Samples Only###
###Outlier SNPs###
neosho_outlier_dapc_plot_black <- ggplot(dapc_neosho_outlier_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = F) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "lightgreen", "sienna4", "lightyellow", "firebrick3","lightsalmon","deepskyblue","mediumpurple","forestgreen","orchid1")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 20)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) +
   theme(axis.text = element_text(color = "white")) +
   theme(axis.title = element_text(color = "white")) 

###Neutral SNPs###
neosho_neutral_dapc_plot_black <- ggplot(dapc_neosho_neutral_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "lightgreen", "sienna4", "lightyellow", "firebrick3","lightsalmon","deepskyblue","mediumpurple","forestgreen","orchid1")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 20, color = "white")) +
   theme(legend.position = c(0.15,0.76)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.title.y = element_blank()) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) +
   theme(axis.text = element_text(color = "white")) +
   theme(axis.title = element_text(color = "white"))

#Combine both DAPC Plots
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/outlier_fst_figures/dapc_plots_black.pdf", width=16, height=14)

plot_grid(smb_outlier_dapc_plot_black, smb_neutral_dapc_plot_black, neosho_outlier_dapc_plot_black, neosho_neutral_dapc_plot_black, nrow = 2, labels = c("a","b","c","d"), label_size = 27)

dev.off()
```

```{r}

plotQ(all_pure_3,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_pure,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GLVR11","GRSPB52","SPRMO49","GRSPB51","GRSPB41","GRSPB37","GRSPB39","GRSPB35","GRSPB36","TBLR02","TBLR03","TBLR04","TBLR01","GRSPB50","SKIA05","SKIA03","SKIA06","SKIA09","SKIA04","STOCK03","STOCK04","STOCK06","STOCK05","BP07","BP17","BP10","BP01","BP02","HC40","HC43","HC44","SC37","SC06","HC38","HC42","HC41","BFORK02","CANEY16","GRSPB69","NOIS12","GRSPB02","GRSPB03","BFORK49","BFORK23","BFC06","BFC49","BFC02","AR18","AR21","AR19","AR16"),
      clustercol = c("deeppink2","goldenrod3","deepskyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/admixture_figures/pure_lineages_3",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)



plotQ(all_pure_4,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_pure,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB51","GRSPB52","GLVR11","SPRMO49","GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK49","GRSPB02","GRSPB03","NOIS12","BFC02","BFC06","BFC49","AR18","AR19","AR21","AR16"),
      clustercol = c("goldenrod3","deeppink2","deepskyblue","navyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/admixture_figures/pure_lineages_4",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)


plotQ(all_pure_5,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_pure,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB51","GRSPB52","GLVR11","SPRMO49","GRSPB36","GRSPB35","GRSPB50","GRSPB41","GRSPB39","GRSPB37","TBLR03","TBLR01","TBLR02","TBLR04","SKIA05","SKIA06","SKIA03","SKIA09","SKIA04","BP17","BP02","BP01","BP07","BP10","STOCK05","STOCK06","STOCK04","STOCK03","HC38","HC41","HC44","HC42","HC40","HC43","SC37","SC06","GRSPB69","BFORK02","BFORK23","CANEY16","BFORK49","GRSPB02","GRSPB03","NOIS12","BFC02","BFC06","BFC49","AR18","AR19","AR21","AR16"),
      clustercol = c("deeppink2","goldenrod3","forestgreen","deepskyblue","navyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/admxixture_figures/pure_lineages_5",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)


```

## Only Northern and Neosho Smallmouth Bass samples (SMB)
```{r}
smb_pure_3 <- readQ("../ADMIXTURE_analysis/admix_all_samples/pass2_pure/smb_pure.3.Q") # K=3 was the best supported value of K at CV = 0.276

plotQ(smb_pure_3,
      showindlab = T, 
      useindlab=T,
      indlabsize = 8,
      indlabcol = "black",
      grplab = genomics_metadata_pure_smb,
      showgrplab = F,
      ordergrp = T,
      selgrp = "sample_id",
      subsetgrp =  c("GRSPB41","GRSPB37","GRSPB39","GRSPB35","GRSPB36","TBLR02","TBLR03","TBLR04","TBLR01","GRSPB50","SKIA05","SKIA03","SKIA06","SKIA09","SKIA04","STOCK03","STOCK04","STOCK06","STOCK05","BP07","BP17","BP10","BP01","BP02","HC40","HC43","HC44","SC37","SC06","HC38","HC42","HC41","BFORK02","CANEY16","GRSPB69","NOIS12","GRSPB02","GRSPB03","BFORK49","BFORK23","BFC06","BFC49","BFC02","AR18","AR21","AR19","AR16"),
      clustercol = c("deeppink2","deepskyblue","navyblue"),
      showlegend = F,
      legendmargin = c(2,2,2,2),
      showsp = F, 
      showdiv = T, 
      divsize = 0.5, 
      divcol = "black",  
      divtype = 1,
      legendkeysize = 10, 
      legendtextsize = 15, 
      linesize = 0.5, 
      pointsize = 4, 
      barbordercolour = "black",
      barbordersize = 0.2, 
      outputfilename = "../../visualization/Admixture/smb_pure_lineages_3",
      imgtype = "pdf",
      imgoutput = "sep",
      height = 5, width = 30)
```
## BLACK AND WHITE plots
```{r}
## HSYC SKIA Plot
dapc_hsyc_skia_admix_LDs <- as.data.frame(dapc_hsyc_skia_admix$ind.coord)
dapc_hsyc_skia_admix_LDs <- cbind(data.frame(admix_pops), dapc_hsyc_skia_admix_LDs)
colnames(dapc_hsyc_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_skia_admix_plot_black <- ggplot(dapc_hsyc_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "grey44","grey100", "firebrick3","grey84","grey64","deepskyblue","mediumpurple","navyblue", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(legend.text = element_text(color = "white")) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_plot_black <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("lightgreen", "grey44","grey100", "sienna4", "lightyellow","lightsalmon","grey84","grey64","deepskyblue","orchid1", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20, color = "white")) +
   theme(axis.text = element_text(size = 20, color = "white")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybriddetective_figures/hyb_simulation_plots_black.pdf", width=9, height=12)

plot_grid(dapc_hsyc_white_admix_plot_black, dapc_hsyc_skia_admix_plot_black, nrow = 2, labels = c("a","b"), label_size = 27)

dev.off()
```