---
title: "SMB_HYBRID_INDEX_AND_CLINES"
author: "Joe Gunn"
date: "11/9/2020"
output: html_document
---

## Libraries needed for analysis
```{r}
library(readxl)
library(tidyverse)
library(adegenet)
library(cowplot)
library(gghybrid)
library(tidyr)
```

## Read in data to gghybrid and prepare for hybrid index analysis - I am basing all of this code on the example code given on the github repository for RIBailey/gghybrid. The data needed to be in genotype format, with one column per locus, two rows per individual containing the two alleles for each locus, and two additional columns at the front of the data set, one for individual id (INDLABEL) and one for population id (POPID)
```{r}

#CANBAR and SKIA
canbar_skia_data <- read.data("../../raw_data/gghybrid_data/canbar_skia_genotypes.csv", 
                              nprecol = 2, 
                              MISSINGVAL = NA, 
                              markername.dup = 0, 
                              INDLABEL = 1, 
                              POPID = 1)

prep_canbar_skia <- data.prep(data = canbar_skia_data$data,
                              loci = canbar_skia_data$loci,
                              alleles = canbar_skia_data$alleles,
                              S0 = "CANBAR",
                              S1 = "SKIA",
                              precols = canbar_skia_data$precols,
                              max.S.MAF = 0.1,
                              return.genotype.table = T,
                              return.locus.table = T)

#CANBAR, HSYC and WHITE
hsyc_canbar_white_data <- read.data("../../raw_data/gghybrid_data/hsyc_canbar_white_genotypes.csv", 
                              nprecol = 2, 
                              MISSINGVAL = NA, 
                              markername.dup = 0, 
                              INDLABEL = 1, 
                              POPID = 1)

prep_hsyc_canbar_white <- data.prep(data = hsyc_canbar_white_data$data,
                              loci = hsyc_canbar_white_data$loci,
                              alleles = hsyc_canbar_white_data$alleles,
                              S0 = "HSYC_CANBAR",
                              S1 = "WHITE",
                              precols = hsyc_canbar_white_data$precols,
                              max.S.MAF = 0.1,
                              return.genotype.table = T,
                              return.locus.table = T)

#HSYC and WHITE
hsyc_white_data <- read.data("../../raw_data/gghybrid_data/hsyc_white_genotypes.csv", 
                              nprecol = 2, 
                              MISSINGVAL = NA, 
                              markername.dup = 0, 
                              INDLABEL = 1, 
                              POPID = 1)

prep_hsyc_white <- data.prep(data = hsyc_white_data$data,
                              loci = hsyc_white_data$loci,
                              alleles = hsyc_white_data$alleles,
                              S0 = "HSYC",
                              S1 = "WHITE",
                              precols = hsyc_white_data$precols,
                              max.S.MAF = 0.1,
                              return.genotype.table = T,
                              return.locus.table = T)
```

## Run Hybrid Index analysis - again, I am using code from the RI Bailey github repository
```{r}
hyb_index_canbar_skia <- esth(prep_canbar_skia$data.prep,
                             read.data.precols = canbar_skia_data$precols,
                             include.Source = T,
                             plot.ind = c("NOIS07","NOIS08","NOIS18","NOIS12","GLVR4","CANEY15"),
                             plot.col = c("blue","green","cyan","purple","magenta","red"),
                             nitt = 10000,
                             burnin = 5000)

hyb_index_hsyc_canbar_white <- esth(prep_hsyc_canbar_white$data.prep,
                             read.data.precols = hsyc_canbar_white_data$precols,
                             include.Source = T,
                             plot.ind = c("AT05","AR29","AT09","AR30","AT08","AT02"),
                             plot.col = c("blue","green","cyan","purple","magenta","red"),
                             nitt = 10000,
                             burnin = 5000)

hyb_index_hsyc_white <- esth(prep_hsyc_white$data.prep,
                             read.data.precols = hsyc_white_data$precols,
                             include.Source = T,
                             plot.ind = c("ER17","ER20","MI419","BC01","SPV11","BC12"),
                             plot.col = c("blue","green","cyan","purple","magenta","red"),
                             nitt = 10000,
                             burnin = 5000)
```

## Plotting 
```{r}
##Hybrid Index Plot for Hsyc, White, Elk, and Uppark populations
hsyc_white_hyb_index_data <- as.data.frame(hyb_index_hsyc_white$hi) 

hsyc_white_hyb_index_data$Source[hsyc_white_hyb_index_data$Source == "TEST"] <- "Hybrids"
hsyc_white_hyb_index_data$Source[hsyc_white_hyb_index_data$Source == "S0"] <- "HSYC"
hsyc_white_hyb_index_data$Source[hsyc_white_hyb_index_data$Source == "S1"] <- "WHITE"

hsyc_white_hyb_index_data <- hsyc_white_hyb_index_data %>%
  mutate(INDLABEL = factor(INDLABEL), POPID = factor(POPID)) %>%
  mutate(INDLABEL = fct_reorder(INDLABEL, beta_mean, .fun = "median"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/genomic_cline_analysis/hsyc_white_hyb_index.pdf", width = 12, height = 7)

ggplot(hsyc_white_hyb_index_data, aes(x = INDLABEL, y = beta_mean, fill = POPID)) + 
  theme_set(theme_cowplot(12)) +
  geom_errorbar(aes(ymin=beta_mean-(beta_mean-h_cred_int_lower), ymax=beta_mean+(h_cred_int_upper-beta_mean), color = POPID)) +
  geom_point(color = "black", size = 3, pch = 21) +
  labs(x = "Individual", y = "Hybrid Index", fill = "Population", color = "Population") +
  facet_wrap(~Source, ncol = 3, scales = "free_x", shrink = F) +
  scale_fill_manual(values = c("sienna4","grey","orchid1","grey"), breaks = c("ELK","UPPARK")) +
  scale_color_manual(values = c("sienna4","grey","orchid1","grey"), breaks = c("ELK","UPPARK")) +
  theme(axis.text.x = element_text(angle = 50)) +
  theme(strip.background = element_rect(fill = "white")) +
  theme(strip.text = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.background = element_rect(color = "black", fill = "white", size = 1, linetype = "solid")) +
  theme(panel.margin = unit(1, "lines")) +
  theme(axis.text.x = element_blank()) +
  theme(axis.ticks = element_blank()) +
  theme(axis.text = element_text(size = 20)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(legend.background = element_rect(fill = "white")) +
  theme(legend.position = c(0.35,0.93)) +
  theme(legend.text = element_text(size = 15)) +
  theme(legend.title = element_blank())

dev.off()

##Hybrid Index Plot for Canbar, Skia, and Illi populations
canbar_skia_hyb_index_data <- as.data.frame(hyb_index_canbar_skia$hi) 

canbar_skia_hyb_index_data$Source[canbar_skia_hyb_index_data$Source == "TEST"] <- "Hybrids"
canbar_skia_hyb_index_data$Source[canbar_skia_hyb_index_data$Source == "S0"] <- "CANBAR"
canbar_skia_hyb_index_data$Source[canbar_skia_hyb_index_data$Source == "S1"] <- "SKIA"

canbar_skia_hyb_index_data <- canbar_skia_hyb_index_data %>%
  mutate(INDLABEL = factor(INDLABEL), POPID = factor(POPID)) %>%
  mutate(INDLABEL = fct_reorder(INDLABEL, beta_mean, .fun = "median"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/genomic_cline_analysis/canbar_skia_hyb_index.pdf", width = 12, height = 7)

ggplot(canbar_skia_hyb_index_data, aes(x = INDLABEL, y = beta_mean, fill = POPID)) + 
  theme_set(theme_cowplot(12)) +
  geom_errorbar(aes(ymin=beta_mean-(beta_mean-h_cred_int_lower), ymax=beta_mean+(h_cred_int_upper-beta_mean), color = POPID)) +
  geom_point(color = "black", size = 3, pch = 21) +
  labs(x = "Individual", y = "Hybrid Index", fill = "Population", color = "Population") +
  facet_wrap(~Source, ncol = 3, scales = "free_x", shrink = F) +
  scale_fill_manual(values = c("grey","mediumpurple","grey"), breaks = c("ILLI")) +
  scale_color_manual(values = c("grey","mediumpurple","grey"),  breaks = c("ILLI")) +
  theme(axis.text.x = element_text(angle = 50)) +
  theme(strip.background = element_rect(fill = "white")) +
  theme(strip.text = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.background = element_rect(color = "black", fill = "white", size = 1, linetype = "solid")) +
  theme(panel.margin = unit(1, "lines")) +
  theme(axis.text.x = element_blank()) +
  theme(axis.ticks = element_blank()) +
  theme(axis.text = element_text(size = 20)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(legend.background = element_rect(fill = "white")) +
  theme(legend.position = c(0.35,0.93)) +
  theme(legend.text = element_text(size = 15)) +
  theme(legend.title = element_blank())

dev.off()


##Hybrid Index Plot for Hsyc, Canbar, and Bayou populations
hsyc_canbar_white_hyb_index_data <- as.data.frame(hyb_index_hsyc_canbar_white$hi) 

hsyc_canbar_white_hyb_index_data$Source[hsyc_canbar_white_hyb_index_data$Source == "TEST"] <- "Hybrids"
hsyc_canbar_white_hyb_index_data$Source[hsyc_canbar_white_hyb_index_data$Source == "S0"] <- "HSYC_CANBAR"
hsyc_canbar_white_hyb_index_data$Source[hsyc_canbar_white_hyb_index_data$Source == "S1"] <- "WHITE"

hsyc_canbar_white_hyb_index_data <- hsyc_canbar_white_hyb_index_data %>%
  mutate(INDLABEL = factor(INDLABEL), POPID = factor(POPID)) %>%
  mutate(INDLABEL = fct_reorder(INDLABEL, beta_mean, .fun = "median"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/genomic_cline_analysis/hsyc_canbar_white_hyb_index.pdf", width = 12, height = 7)

ggplot(canbar_skia_hyb_index_data, aes(x = INDLABEL, y = beta_mean, fill = POPID)) + 
  theme_set(theme_cowplot(12)) +
  geom_errorbar(aes(ymin=beta_mean-(beta_mean-h_cred_int_lower), ymax=beta_mean+(h_cred_int_upper-beta_mean), color = POPID)) +
  geom_point(color = "black", size = 3, pch = 21) +
  labs(x = "Individual", y = "Hybrid Index", fill = "Population", color = "Population") +
  facet_wrap(~Source, ncol = 3, scales = "free_x", shrink = F) +
  scale_fill_manual(values = c("grey","lightgreen","grey"), breaks = c("BAYOU")) +
  scale_color_manual(values = c("grey","lightgreen","grey"),  breaks = c("BAYOU")) +
  theme(axis.text.x = element_text(angle = 50)) +
  theme(strip.background = element_rect(fill = "white")) +
  theme(strip.text = element_text(size = 20, hjust = 0.5, face = "bold")) +
  theme(panel.background = element_rect(color = "black", fill = "white", size = 1, linetype = "solid")) +
  theme(panel.margin = unit(1, "lines")) +
  theme(axis.text.x = element_blank()) +
  theme(axis.ticks = element_blank()) +
  theme(axis.text = element_text(size = 20)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(legend.background = element_rect(fill = "white")) +
  theme(legend.position = c(0.35,0.93)) +
  theme(legend.text = element_text(size = 15)) +
  theme(legend.title = element_blank())

dev.off()
```

```{r}

cline_hsyc_white <- ggcline(
    data.prep.object = prep_hsyc_white$data.prep,
    esth.object = hyb_index_hsyc_white,
    read.data.precols = hsyc_white_data$precols,
    nitt = 10000, burnin = 5000, print.k = 50)

```