---
title: "SMB_HYBRID_DETECTIVE"
author: "Joe Gunn"
date: "7/10/2020"
output: html_document
---

## Libraries needed for analysis
```{r}
library(genepopedit)
library(parallelnewhybrid)
library(hybriddetective)
library(readxl)
library(tidyverse)
library(adegenet)
library(cowplot)
```

## Prepare parent genepop datasets for PLINK (I used pgdspider on my computer to convert vcf files for only the parent populations, MIDARK and WHITE, and MIDARK and SKIA, into genepop files. The genepop files created in pgdspider do not have the correct population designations, so they need to be reformatted with the "subset_genepop_rename" code below).
```{r}
#create a dataframe to designate two different populations (Pure Neosho, Pure Northern)
pop_rename_midark_white <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE","WHITE"))

pop_rename_midark_skia <- data.frame(Opop = c("","","","","","","","","","","","","","","","","","","","",""), Rename = c("MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","MIDARK","SKIA","SKIA","SKIA","SKIA","SKIA"))


#Designate the two different populations (Pure Neosho, Pure Northern)
subset_genepop_rename("../../raw_data/hybriddetective_data/raw_parent_genepops/MIDARK_WHITE_parent_genepop_snps_random.txt", pop_rename_midark_white, path = "/Users/joegunn/Desktop/MIDARK_WHITE_edited_parent_genepop_snps_random.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/raw_parent_genepops/MIDARK_SKIA_parent_genepop_snps_random.txt", pop_rename_midark_skia, path = "/Users/joegunn/Desktop/MIDARK_SKIA_edited_parent_genepop_snps_random.txt")


#Explore genepop data using genepopedit package - here, I am just seeing what the data tables look like with respect to the genepop edit package. I am then using the genepopedit pacakge to make sure that all of the genepop files are in the correct format for the hybriddetective getTopLoc() function
genepop_flatten("../../raw_data/hybriddetective_data/edited_parent_genepops/MIDARK_SKIA_edited_parent_genepop_snps_random.txt")
```
## Get top 200 diagnostic loci

## Run getTopLoc() funtion from hybriddetective to get 200 diagnostic, high linkage, high Fst SNPs. There must be a "/" after the directories that PLINK and PGDSpider are in! Otherwise the code won't work and you'll be very confused.
```{r}
#Get panel SNP panel for MIDARK and WHITE populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/edited_parent_genepops/MIDARK_WHITE_edited_parent_genepop.txt",
          panel.size = 200,
          where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/",
          where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")

#Get panel SNP panel for MIDARK and SKIA populations
getTopLoc(GPD = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/edited_parent_genepops/MIDARK_SKIA_edited_parent_genepop.txt", 
          panel.size = 200,
          where.PGDspider = "/Users/joegunn/Downloads/PGDSpider_2.1.1.5/", 
          where.PLINK = "/Users/joegunn/Downloads/plink_mac_20200616/")
```
```{r}

#Simulate hybrid genotypes based on the VALIDATION parent samples (the non-training parent samples - those not used to discover the 200 most informative, diagnostic loci)
freqbasedsim_AlleleSample(GPD="../../raw_data/hybriddetective_data/parent_200_loci_panels/MIDARK_SKIA_edited_parent_genepop_200_Loci_Panel.txt", NumSims = 3, NumReps = 3)

#Run parallelnewhybrids program to assign simulated genotypes to hybrid class (Bayesian Gibbs sampling algorithm)
parallelnh_OSX("../../raw_data/hybriddetective_data/sim_newhybrid_panels/", where.NH = "/Users/joegunn/Downloads/newhybrids/", burnin = 100, sweeps = 500)

#Check for NewHybrids model convergence - this code will generate a message that either says to re-check the results (the Bayesian threads did not converge) or that you are good to go (the threads successfully converged). Failure to converge results in most individuals being assigned erroneously to the "F2" hybrid class.
nh_preCheckR(PreDir="/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/sim_newhybrid_panels/NH.Results/")

#Plot results for NewHybrids model convergence
nh_multiplotR(NHResults = "/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/sim_newhybrid_panels/NH.Results/")

getwd()
```

## Determine the SNP names for the 200 diagnostic SNPs discovered in the above analyses - this is necessary, because the hybriddetective workflow code just generates the number of the SNPs that were used to make the panel (SNP_1, 2, 3, etc.) rather than the actual SNP name.
```{r}
snp_metadata <- read_excel("../../metadata/smb_snp_metadata.xlsx")
snp_metadata <- snp_metadata %>%
  mutate(snp_number = factor(snp_number))

#HSYC with the White River SNPs
hsyc_white_200_panel_snps <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_200_panels/hsyc_white_200_loci_panel.xlsx")
hsyc_white_200_panel_snps <- hsyc_white_200_panel_snps %>%
  mutate(snp_number = factor(snp_number))

hsyc_white_200_panel_snp_names <- merge(snp_metadata, hsyc_white_200_panel_snps, by = "snp_number")


#CANBAR with Skiatook Lake SNPs
canbar_skia_200_panel_snps <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_200_panels/canbar_skia_200_loci_panel.xlsx")
canbar_skia_200_panel_snps <- canbar_skia_200_panel_snps %>%
  mutate(snp_number = factor(snp_number))

canbar_skia_200_panel_snp_names <- merge(snp_metadata, canbar_skia_200_panel_snps, by = "snp_number")


#CANBAR + HSYC with White River SNPs
hsyc_canbar_white_200_panel_snps <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_200_panels/hsyc_canbar_white_200_loci_panel.xlsx")
hsyc_canbar_white_200_panel_snps <- hsyc_canbar_white_200_panel_snps %>%
  mutate(snp_number = factor(snp_number))

hsyc_canbar_white_200_panel_snp_names <- merge(snp_metadata, hsyc_canbar_white_200_panel_snps, by = "snp_number")


#Write out tables with SNP names to be kept for hybrid analysis - these will be text files that can be later used in vcftools to subset vcfs on Lewis 
write.table(hsyc_white_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_200_panels/hsyc_white_200_panel_snp_names.txt", sep="\t")

write.table(canbar_skia_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_200_panels/canbar_skia_200_panel_snp_names.txt", sep="\t")

write.table(hsyc_canbar_white_200_panel_snp_names,"~/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/raw_data/hybriddetective_data/hyb_detective_200_panels/hsyc_canbar_white_200_panel_snp_names.txt", sep="\t")
```

## HYBRIDLAB ANALYSIS
### In the program HYBRIDLAB, I am simulating 6 hybrid categories (P1, P2, F1, F2, BC1, BC2) for hybrid populations between HSYC and WHITE and between HSYC and SKIA. I am generating 100 simulated individuals for each category, including simulated parent individuals, even though I have empirical data for the parents. This will allow me to assess all differences or similarities with equal sample sizes. Simulated genotypes will be compared to the individual admixed populations from my study

## Format simulated data files from HYBRIDLAB 
```{r}

#Designate the two different populations (Pure Neosho, Pure Northern) for SIMULATED DATASETS
subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_parent_sim/hsyc_white_parent_sim.txt", pop_rename_hsyc_white, path = "/Users/joegunn/Desktop/hsyc_white_parent_sim_genepop_goodformat.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_parent_sim/canbar_skia_parent_sim.txt", pop_rename_canbar_skia, path = "/Users/joegunn/Desktop/canbar_skia_parent_sim_genepop_goodformat.txt")

subset_genepop_rename("../../raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_parent_sim/hsyc_canbar_white_parent_sim.txt", pop_rename_hsyc_canbar_white, path = "/Users/joegunn/Desktop/hsyc_canbar_white_parent_sim_genepop_goodformat.txt")

#Take a look at the datafiles to see if they are in the correct format
genepop_detective("../../raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_parent_sim/hsyc_white_parent_sim_genepop_goodformat.txt", variable = "PopNum")
genepop_flatten("../../raw_data/hybriddetective_data/hyb_detective_genepops/hyb_detective_parent_sim/hsyc_canbar_white_parent_sim_genepop_goodformat.txt")
```

## DAPC plotting for hybridlab simulations
### In order to convert the simulated genepop data into a genind formatted file for DAPC in adegenet, I had to create a matrix of genotypes, rather than using a standard genepop format (I think there is a way to use a genepop file strictly, but I couldn't figure it out). My genotype matrix consisted of one row per individual sample, and each column was a SNP (200 total). Genotypes were coded as ###### (six numbers in a row)
```{r}
## Run DAPC for HSYC and SKIA parents, along with admixed populations (BARON, CANEY, ILLINOIS)
canbar_skia_all_admix_genepop <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_sim_hybrids/canbar_skia_ALL_SIM_hybrids.xlsx")

## Create population vector
admix_pops_canbar_skia <- factor(c(rep("Simulated CANBAR", times = 100), rep("Simulated SKIA", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("Backcross to CANBAR", times = 100), rep("Backcross to SKIA", times = 100), rep("ILLI", times = 8)))

canbar_skia_all_admix_genepop <- column_to_rownames(canbar_skia_all_admix_genepop, "ind")
canbar_skia_all_admix_genepop <- as.matrix(canbar_skia_all_admix_genepop)

## Get genind file
canbar_skia_all_admix_genepop_genind <- df2genind(canbar_skia_all_admix_genepop, 
                                                ncode = 3, 
                                                pop = admix_pops_canbar_skia, 
                                                ploidy = 2)
## Scale genind file
canbar_skia_all_admix_genepop_genind <- scaleGen(canbar_skia_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

## Run DAPC
dapc_canbar_skia_admix <- dapc(canbar_skia_all_admix_genepop_genind, admix_pops_canbar_skia)

## PCs retained: 10
## Discriminant Functions retained: 3

## Run DAPC for HSYC and WHITE parents, along with admixed populations (SPAV, ELK, BSUG, BUFF, BAYOU)
hsyc_white_all_admix_genepop <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_sim_hybrids/hsyc_white_ALL_SIM_hybrids.xlsx")

## Create population vector
admix_pops_hsyc_white <- factor(c(rep("Simulated HSYC", times = 100), rep("Simulated WHITE", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("Backcross to HSYC", times = 100), rep("Backcross to WHITE", times = 100), rep("ELK", times = 6), rep("UPPARK", times = 18)))

hsyc_white_all_admix_genepop <- column_to_rownames(hsyc_white_all_admix_genepop, "pop")
hsyc_white_all_admix_genepop <- as.matrix(hsyc_white_all_admix_genepop)

## Get genind file
hsyc_white_all_admix_genepop_genind <- df2genind(hsyc_white_all_admix_genepop, 
                                                 ncode = 3, 
                                                 pop = admix_pops_hsyc_white, 
                                                 ploidy = 2)
## Scale genind file
hsyc_white_all_admix_genepop_genind <- scaleGen(hsyc_white_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

# Run DAPC
dapc_hsyc_white_admix <- dapc(hsyc_white_all_admix_genepop_genind, admix_pops_hsyc_white)

## PCs retained: 10
## Discriminant Functions retained: 3

## Run DAPC for HSYC and WHITE parents, along with admixed populations (SPAV, ELK, BSUG, BUFF, BAYOU)
hsyc_canbar_white_all_admix_genepop <- read_excel("../../raw_data/hybriddetective_data/hyb_detective_sim_hybrids/hsyc_canbar_white_ALL_SIM_hybrids.xlsx")

## Create population vector
admix_pops_hsyc_canbar_white <- factor(c(rep("Simulated HSYC+CANBAR", times = 100), rep("Simulated WHITE", times = 100), rep("F1", times = 100), rep("F2", times = 100), rep("Backcross to HSYC+CANBAR", times = 100), rep("Backcross to WHITE", times = 100), rep("BAYOU", times = 7)))

hsyc_canbar_white_all_admix_genepop <- column_to_rownames(hsyc_canbar_white_all_admix_genepop, "ind")
hsyc_canbar_white_all_admix_genepop <- as.matrix(hsyc_canbar_white_all_admix_genepop)

## Get genind file
hsyc_canbar_white_all_admix_genepop_genind <- df2genind(hsyc_canbar_white_all_admix_genepop, 
                                                 ncode = 3, 
                                                 pop = admix_pops_hsyc_canbar_white, 
                                                 ploidy = 2)
## Scale genind file
hsyc_canbar_white_all_admix_genepop_genind <- scaleGen(hsyc_canbar_white_all_admix_genepop_genind, 
                                           center = TRUE, 
                                           scale = TRUE, 
                                           NA.method = c("mean"), 
                                           truenames = TRUE)

# Run DAPC
dapc_hsyc_canbar_white_admix <- dapc(hsyc_canbar_white_all_admix_genepop_genind, admix_pops_hsyc_canbar_white)

## PCs retained: 10
## Discriminant Functions retained: 3

```

## Make plots
```{r}
## HSYC SKIA Plot
dapc_canbar_skia_admix_LDs <- as.data.frame(dapc_canbar_skia_admix$ind.coord)
dapc_canbar_skia_admix_LDs <- cbind(data.frame(admix_pops_canbar_skia), dapc_canbar_skia_admix_LDs)
colnames(dapc_canbar_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_canbar_skia_admix_LDs$pop <- factor(dapc_canbar_skia_admix_LDs$pop, levels = c("Simulated CANBAR","Backcross to CANBAR", "F2", "F1", "Backcross to SKIA","Simulated SKIA", "ILLI"))

dapc_canbar_skia_admix_plot <- ggplot(dapc_canbar_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Hybrid Category") +
   scale_fill_manual(values = c("grey100","grey80","grey60","grey40","grey20","black","mediumpurple")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.6,0.8)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.title.y = element_blank())

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_hsyc_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_LDs$pop <- factor(dapc_hsyc_white_admix_LDs$pop, levels = c("Simulated HSYC", "Backcross to HSYC","F2","F1","Empirical HSYC","Empirical WHITE","Backcross to WHITE","Simulated WHITE","ELK","UPPARK"))

dapc_hsyc_white_admix_plot <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Hybrid Category") +
   scale_fill_manual(values = c("black","grey20","grey40","grey60","grey80", "grey100", "sienna4", "orchid1")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.05,0.8)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) 

## HSYC WHITE Plot
dapc_hsyc_canbar_white_admix_LDs <- as.data.frame(dapc_hsyc_canbar_white_admix$ind.coord)
dapc_hsyc_canbar_white_admix_LDs <- cbind(data.frame(admix_pops_hsyc_canbar_white), dapc_hsyc_canbar_white_admix_LDs)
colnames(dapc_hsyc_canbar_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_canbar_white_admix_LDs$pop <- factor(dapc_hsyc_canbar_white_admix_LDs$pop, levels = c("Simulated HSYC+CANBAR","Backcross to HSYC+CANBAR","F2","F1","Backcross to WHITE","Simulated WHITE", "BAYOU"))

dapc_hsyc_canbar_white_admix_plot <- ggplot(dapc_hsyc_canbar_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Hybrid Category") +
   scale_fill_manual(values = c("black", "grey20","grey40","grey60","grey80", "grey100", "lightgreen")) +
   theme_set(theme_cowplot(12)) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold")) +
   theme(legend.text = element_text(size = 15)) +
   theme(legend.position = c(0.05,0.8)) +
   theme(legend.background = element_rect(color = "black", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.title.y = element_blank())


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybrid_detective/hyb_simulation_plots.pdf", width=25, height=8)

plot_grid(dapc_hsyc_white_admix_plot, dapc_canbar_skia_admix_plot, dapc_hsyc_canbar_white_admix_plot, nrow = 1, labels = c("a","b","c"), label_size = 27)

dev.off()
```

## BLACK AND WHITE plots
```{r}
## HSYC SKIA Plot
dapc_hsyc_skia_admix_LDs <- as.data.frame(dapc_hsyc_skia_admix$ind.coord)
dapc_hsyc_skia_admix_LDs <- cbind(data.frame(admix_pops), dapc_hsyc_skia_admix_LDs)
colnames(dapc_hsyc_skia_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_skia_admix_plot_black <- ggplot(dapc_hsyc_skia_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("turquoise", "grey44","grey100", "firebrick3","grey84","grey64","deepskyblue","mediumpurple","navyblue", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(legend.text = element_text(color = "white")) +
   theme(axis.title = element_text(size = 20)) +
   theme(axis.text = element_text(size = 20)) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

## HSYC WHITE Plot
dapc_hsyc_white_admix_LDs <- as.data.frame(dapc_hsyc_white_admix$ind.coord)
dapc_hsyc_white_admix_LDs <- cbind(data.frame(admix_pops_white), dapc_hsyc_white_admix_LDs)
colnames(dapc_hsyc_white_admix_LDs) <- c("pop","LD1","LD2")

dapc_hsyc_white_admix_plot_black <- ggplot(dapc_hsyc_white_admix_LDs, aes(x = LD1, y = LD2, fill = pop)) +
   geom_point(aes(fill = pop), color = "black", pch = 21, size = 5, show.legend = T) +
   labs(x = "LD1", y = "LD2", fill = "Population") +
   scale_fill_manual(values = c("lightgreen", "grey44","grey100", "sienna4", "lightyellow","lightsalmon","grey84","grey64","deepskyblue","orchid1", "deeppink2")) +
   theme_set(theme_cowplot(12)) +
   theme(plot.background = element_rect(fill = "black")) +
   theme(legend.title = element_text(size = 20)) +
   theme(legend.title = element_text(face = "bold", color = "white")) +
   theme(legend.text = element_text(size = 15, color = "white")) +
   theme(legend.position = c(0.7,0.7)) +
   theme(legend.background = element_rect(color = "white", size = 0.5)) + 
   theme(legend.margin = margin(5, 5, 5, 5)) +
   theme(axis.title = element_text(size = 20, color = "white")) +
   theme(axis.text = element_text(size = 20, color = "white")) +
   theme(axis.line = element_line(color = "white")) +
   theme(axis.ticks = element_line(color = "white")) + 
   theme(axis.title = element_text(color = "white")) +
   theme(axis.text = element_text(color = "white"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/hybrid_detective/hyb_simulation_plots_black.pdf", width=9, height=12)

plot_grid(dapc_hsyc_white_admix_plot_black, dapc_hsyc_skia_admix_plot_black, nrow = 2, labels = c("a","b"), label_size = 27)

dev.off()
```
