---
title: "SMB_ALLELE_FREQUENCIES"
author: "Joe Gunn"
date: "12/11/2019"
output: html_document
---

## Libraries needed for Analysis
```{r libraries, echo = FALSE, include=FALSE}
library(readxl)
library(tidyverse)
library(cowplot)
library(devtools)
library(SNPRelate)
library(vcfR)
library(pophelper)
library(PopGenome)
library(pcadapt)
library(ggforce)
```

## Preliminary Data Analyses
### All preliminary filtering steps were conducted sequentially on VCF files in VCFTOOLS. Outputs from VCFTOOLS were converted into SNP tables (easily readable files which include total number of homozygotes and heterozgyotes of reference and alternative alleles) with GATK and read into Excel. Excel files were then manipulated and analyzed in R.

## Data read-in and cleaning for MISSING DATA (file output from VCFTOOLS: out.imiss)
```{r}
#Read in raw data for missing genotype calls per sample. This file gives the percentage of missing genotype calls (F_MISS) across all SNPs in the original, all-filtered vcf file (50,828 SNPs). This file was produced with VCFTools using the --missing-indv flag. The output file is called out.imiss
missing_data <- read_excel("../../raw_data/allele_frequency_data/smb_genomics_missingdata_bysample.xlsx")
```

## Plot Missing Data per Sample
```{r}
#Excel file with proportion of missing data by sample. All samples are included

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/preliminary_filtering_figures/missing_data_plot.pdf", width=7, height=25)

ggplot(missing_data, aes(x = sample_id, y = F_MISS)) + 
  geom_bar(position=position_dodge(), stat="identity", show.legend = T, fill = "grey", color = "black") + 
  ylim(0,1) +
  geom_hline(yintercept = 0.2, linetype = "longdash", color = "red", size = 2) + 
  theme_set(theme_cowplot(12)) + 
  labs(x = "Sample ID", y = "Proportion of Missing Data") + 
  coord_flip() + 
  theme(axis.text.x = element_text(angle = 60, hjust = 1)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(axis.text = element_text(size = 20))

dev.off()
```

## Data read-in and cleaning for ALLELE AND GENOTYPE FREQUENCIES
```{r}
#Genotype frequencies, allele frequencies, and read depth were all analyzed at sequential steps of the filtering process. The number of samples and number of SNPs included in each datafile are explicitly stated next to the read-in for that datafile.

##After filtering for poor quality samples (more than 20% missing data) and poor phred quality samples (phred quality score less than 20): 127,024 SNPs, 92 samples
snp_table <- read_excel("../../raw_data/allele_frequency_data/vcf_badsamples_qual.xlsx")
snp_table <- snp_table %>% mutate(CHROM = as.character(CHROM))
snp_table <- snp_table %>% mutate(CHROM = as.factor(CHROM))

##After filtering for por quality samples, poor phred score, and thinning to one SNP per contig: 65,254 SNPs, 92 samples
snp_table_onerad <- read_excel("../../raw_data/allele_frequency_data/vcf_badsamples_qual_onerad.xlsx")
snp_table_onerad <- snp_table_onerad %>% mutate(CHROM = as.character(CHROM))
snp_table_onerad <- snp_table_onerad %>% mutate(CHROM = as.factor(CHROM))

##After filtering for por quality samples, poor phred score, thinning to one SNP per contig, missing data across SNPs greater than 20%, minor allele count of 2 (maf = 0.011 <--- 2/(92 X 2)), and greater than 45 percent heterozygosity. For some reason I used the term  "het50" in the file name to indicate greater than 45% heterozygosity, but I confirmed that I did exclude all SNPs with greater than 45% heterozygosity.
snp_table_filtered <- read_excel("../../raw_data/allele_frequency_data/vcf_badsamples_qual_onerad_missing_mac2_het50.xlsx")
snp_table_filtered <- snp_table_filtered %>% mutate(CHROM = as.character(CHROM))
snp_table_filtered <- snp_table_filtered %>% mutate(CHROM = as.factor(CHROM))
```


## Plot Read Depth distributions across all loci BEFORE and 
```{r}
#Calculate mean read depths for all samples BEFORE and AFTER filtering
snp_table_onerad %>% summarize(mean_dp = mean(DP)) #mean read-depth before filtering (using only one SNP per RAD tag) is 4393.708
snp_table_filtered %>% summarize(mean_dp = mean(DP)) #mean read-depth after filtering (using only one SNP per RAD tag) is 4171.364

options(scipen = 999)

dp_prefilter <- ggplot(snp_table_onerad, aes(x = DP)) + 
  geom_histogram(binwidth = 10000, fill = "grey", color = "black") + 
  geom_vline(xintercept = 4393.708, color = "black", size = 1) + 
  facet_zoom(ylim = c(0,15)) + 
  theme(axis.text.x = element_text(angle = 30, hjust = 0.9)) + 
  labs(y = "Number of Contigs") + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(axis.title.x = element_blank())

dp_postfilter <- ggplot(snp_table_filtered, aes(x = DP)) + 
  geom_histogram(binwidth = 5000, fill = "grey", color = "black") + 
  geom_vline(xintercept = 4171.364, color = "black", size = 1) + 
  facet_zoom(ylim = c(0,15)) + 
  theme(axis.text.x = element_text(angle = 30, hjust = 0.9)) + 
  labs(x = "Read Depth", y = "Number of Contigs") + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20))


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/preliminary_filtering_figures/mean_read_depth_plot.pdf", width=14, height=10)

plot_grid(dp_prefilter, dp_postfilter, nrow = 2, labels = c("a","b"), label_size = 25)

dev.off()
```

## Minor Allele Frequency Spectrum
```{r}
#The minor allele frequency spectrum was calculated after I applied three filters: omitting samples with greater than 20% missing data, omitting SNPs with phred quality scores less than 20, and omitting all but one SNP per RAD tag. I had also done the minor allele frequency spectrum before all of these filters, but the histograms are almost identical in shape.

#Pre-filtered minor allele frequency spectrum: 65,254 SNPs, 92 samples
#Post-filtered minor allele frequency spectrum: 50,828 SNPs, 92 samples
 
###IMPORTANT NOTE: I am calculating the folded minor allele frequency spectrum for all loci, which includes all minor alleles rather than just the alternate allele (which, in this case, is usually the minor allele). The folded allele frequency spectrum therefore considers the reference alleles that are minor alleles (less than 0.5 in frequency relative to the alternative allele) and the alternative alleles that are minor alleles (less than 0.5 frequency relative to the reference allele).

###PRE-FILTERING###########

#Gather SNPs in which the reference allele is the minor allele
maf_ref_onerad <- snp_table_onerad %>% 
  filter(freq_ref < 0.5) #There are 3,929 reference minor alleles

#Gather SNPs in which the alternative allele is the minor allele
maf_alt_onerad  <- snp_table_onerad %>% 
  filter(freq_alt <= 0.5) #There are 61,325 alternative minor alleles

###CHECK: 3,929 + 61,325 = 65,254 (TOTAL NUMBER OF SNPs in dataset)

#Clean data for later graphing
maf_ref_values_onerad  <- as.data.frame(cbind(maf_ref_onerad$freq_ref, maf_ref_onerad$QUAL, maf_ref_onerad$freq_het))
colnames(maf_ref_values_onerad) <- c("maf", "qual", "het")

maf_alt_values_onerad  <- as.data.frame(cbind(maf_alt_onerad$freq_alt, maf_alt_onerad$QUAL, maf_alt_onerad$freq_het))
colnames(maf_alt_values_onerad) <- c("maf", "qual", "het")

maf_values_onerad <- rbind(maf_ref_values_onerad, maf_alt_values_onerad)

#Calculate the average minor allele frequency pre-filtering
maf_values_onerad %>% 
  summarize(mean(maf_values_onerad$maf)) #Average maf = 0.089


###POST-FILTERING###########


#Gather SNPs in which the reference allele is the minor allele
maf_ref_filtered <- snp_table_filtered %>% 
  filter(freq_ref < 0.5) #There are 3,530 reference minor alleles

#Gather SNPs in which the reference allele is the minor allele
maf_alt_filtered  <- snp_table_filtered %>% 
  filter(freq_alt <= 0.5) #There are 47,298 alternative minor alleles

#Clean data for later graphing
maf_ref_values_filtered  <- as.data.frame(cbind(maf_ref_filtered$freq_ref, maf_ref_filtered$QUAL, maf_ref_filtered$freq_het))
colnames(maf_ref_values_filtered) <- c("maf", "qual", "het")

maf_alt_values_filtered  <- as.data.frame(cbind(maf_alt_filtered$freq_alt, maf_alt_filtered$QUAL, maf_alt_filtered$freq_het))
colnames(maf_alt_values_filtered) <- c("maf", "qual", "het")
maf_values_filtered <- rbind(maf_ref_values_filtered, maf_alt_values_filtered)

#Calculate the average minor allele frequency post-filtering
maf_values_filtered %>% 
  summarize(mean(maf_values_filtered$maf)) #Average maf = 0.106


#Clean and organize pre-filtering and post-filtering datasets for graphing
pre_filter_label <- as.data.frame(c(rep("pre-filtering", times = 65254)))
post_filter_label <- as.data.frame(c(rep("post-filtering", times = 50828)))

maf_values_onerad <- cbind(maf_values_onerad, pre_filter_label)
colnames(maf_values_onerad) <- c("maf", "qual", "het", "filter")

maf_values_filtered <- cbind(maf_values_filtered, post_filter_label)
colnames(maf_values_filtered) <- c("maf", "qual", "het", "filter")

maf_before_after <- rbind(maf_values_onerad, maf_values_filtered)


#Folded minor allele frequency spectrum - pre-filtering
maf_onerad_plot <- ggplot(maf_values_onerad, aes(x = maf)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.08987174, color = "black", size = 2) + 
  geom_vline(xintercept = 0.01086957, color = "red", linetype = "longdash", size = 2) + ##this x-intercept represents my mac cutoff (2)
  theme_set(theme_cowplot(12)) + 
  labs(x = "Minor Allele Frequency", y = "Number of SNPs") + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20))

#Folded minor allele frequency spectrum - post-filtering
maf_filtered_plot <- ggplot(maf_values_filtered, aes(x = maf)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") +
  geom_vline(xintercept = 0.1057535, color = "black", size = 2) +
  theme_set(theme_cowplot(12)) + 
  labs(x = "Minor Allele Frequency", y = "Number of SNPs") + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(axis.title.y = element_blank())


#Pre and post-filtering plots separate
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/preliminary_filtering_figures/minor_allele_frequency_plots.pdf", width=15, height=6)

plot_grid(maf_onerad_plot, maf_filtered_plot, nrow = 1, labels = c("a","b"), label_size = 25)

dev.off()
```

## Check for Paralogs - Alternate Allele Frequency vs. Locus-wise heterozygosity
```{r}
#Here, I am filtering out all SNPs with greater than 45% heterozygosity to reduce the potential for bias due to paralogous loci (duplicate loci that will increase the number of heterozygous loci spuriously)

#filter out all SNPs with greater than 45% heterozgyosity
snp_table_onerad_het45 <- snp_table_onerad %>% 
  filter(freq_het >= 0.45)

#filter out all SNPs with less than 45% heterozgyosity
snp_table_onerad_hetnorm <- snp_table_onerad %>% 
  filter(freq_het < 0.45)

#Clean data and organize for later graphing
het45_lab <- as.data.frame(c(rep("het45", times = 961)))
colnames(het45_lab) <- c("het_lab")

hetnorm_lab <- as.data.frame(c(rep("hetnorm", times = 64293)))
colnames(hetnorm_lab) <- c("het_lab")

het45 <- cbind(snp_table_onerad_het45, het45_lab)
hetnorm <- cbind(snp_table_onerad_hetnorm, hetnorm_lab)

het_table <- rbind(het45, hetnorm)

#Plot heterozygosity against the alternative allele frequency to see the distribution of heterozgotes dependent on alternative allele frequency
het_prefilter <- ggplot(het_table, aes(x = freq_alt, y = freq_het, color = het_lab)) + 
  geom_point(alpha = 0.4, show.legend = F) + 
  geom_hline(yintercept = 0.45, color = "red", linetype = "longdash", size = 2) + 
  scale_color_manual(values = c("red", "black")) + 
  labs(x = "Frequency of Alternate Allele", y = "Frequency") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  theme(axis.title.x = element_blank())

#Plot heterozygosity against the alternative allele frequency to see the distribution of heterozgotes dependent on alternative allele frequency - This time WITHOUT SNPs with over 45% heterozygosity.
het_postfilter <- ggplot(snp_table_filtered, aes(x = freq_alt, y = freq_het)) + 
  geom_point(alpha = 0.4, show.legend = F) + 
  labs(x = "Frequency of Alternate Allele", y = "Frequency") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 20)) + 
  theme(axis.title = element_text(size = 20)) + 
  ylim(c(0,1)) 


pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/preliminary_filtering_figures/alt_het_plots.pdf", width=6, height=9)

plot_grid(het_prefilter, het_postfilter, nrow = 2, labels = c("a", "b"), label_size = 25)

dev.off()
```

# Summarize Allele and Genotype Frequencies across all SNPs BEFORE AND AFTER filtering
## Before Filtering
```{r}
#Here I am plotting allele frequency and genotype frequency distributions across all SNPs and all samples pre-filtering

#mean alternative allele frequency 
snp_table %>% 
  summarize(mean_alt = mean(freq_alt)) #mean alternative allele frequency = 0.012

#mean reference allele frequency 
snp_table %>% 
  summarize(mean_ref = mean(freq_ref)) #mean reference allele frequency = 0.880

#mean heterozygous genotype frequency 
snp_table %>% 
  summarize(mean_het = mean(freq_het)) #mean heterozygous genotype frequency = 0.099

#mean homozygous reference genotype frequency 
snp_table %>% 
  summarize(mean_homo_ref = mean(freq_homo_ref)) #mean homozygous reference genotype frequency = 0.830

#mean homozygous alternative genotype frequency 
snp_table %>% 
  summarize(mean_homo_alt = mean(freq_homo_alt)) #mean homozgous alternative genotype frequency = 0.070
```

## Plot Allele and Genotype Frequencies
```{r}
ref_histogram <- ggplot(snp_table, aes(x = freq_ref)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.879912815, color = "black", size = 2) + 
  labs(title = "Pre-Filtering", x = "Frequency", y = "Count") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

alt_histogram <- ggplot(snp_table, aes(x = freq_alt)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.120087185, color = "black", size = 2) + 
  labs(x = "Frequency ", y = "Count") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

homo_ref_histogram <- ggplot(snp_table, aes(x = freq_homo_ref)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.830017799, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Count") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  ylim(0,24000) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

homo_alt_histogram <- ggplot(snp_table, aes(x = freq_homo_alt)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.070192169, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Count") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) +
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

het_histogram <- ggplot(snp_table, aes(x = freq_het)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.099790033, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Count") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.text.y = element_text(angle = 40))
```

##After filtering
```{r}
#Here I am plotting allele frequency and genotype frequency distributions across all SNPs and all samples post-filtering

#mean alternative allele frequency 
snp_table_filtered %>% 
  summarize(mean_alt = mean(freq_alt)) #mean alternative allele frequency = 0.135

#mean reference allele frequency 
snp_table_filtered %>%
  summarize(mean_ref = mean(freq_ref)) #mean reference allele frequency = 0.864

#mean heterozygous genotype frequency 
snp_table_filtered %>% 
  summarize(mean_het = mean(freq_het)) #mean heterozygous genotype frequency = 0.096

#mean homozygous reference genotype frequency 
snp_table_filtered %>% 
  summarize(mean_homo_ref = mean(freq_homo_ref)) #mean homozygous reference genotype frequency = 0.817

#mean homozygous alternative genotype frequency 
snp_table_filtered %>% 
  summarize(mean_homo_alt = mean(freq_homo_alt)) #mean homozgyous alternative genotype frequency = 0.087
```

## Plot Allele and Genotype Frequencies
```{r}
ref_histogram_postfilter <- ggplot(snp_table_filtered, aes(x = freq_ref)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.8648376, color = "black", size = 2) + 
  labs(title = "Post-Filtering", x = "Frequency", y = "Number of SNPs") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) +
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

alt_histogram_postfilter <- ggplot(snp_table_filtered, aes(x = freq_alt)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.1351624, color = "black", size = 2) + 
  labs(x = "Frequency ", y = "Number of SNPs") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(axis.title.y = element_blank()) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

homo_ref_histogram_postfilter <- ggplot(snp_table_filtered, aes(x = freq_homo_ref)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.8168621, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Number of SNPs") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25),) + 
  theme(axis.title = element_text(size = 25)) +
  theme(axis.title.y = element_blank()) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  ylim(0,24000) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))
 
homo_alt_histogram_postfilter <- ggplot(snp_table_filtered, aes(x = freq_homo_alt)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.08718688, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Number of SNPs") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(axis.title.y = element_blank()) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.x = element_blank()) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))

het_histogram_postfilter <- ggplot(snp_table_filtered, aes(x = freq_het)) + 
  geom_histogram(binwidth = 0.01, fill = "grey", color = "black") + 
  geom_vline(xintercept = 0.09595098, color = "black", size = 2) + 
  labs(x = "Frequency", y = "Number of SNPs") + 
  theme_set(theme_cowplot(12)) + 
  theme(axis.text = element_text(size = 25)) + 
  theme(axis.title = element_text(size = 25)) + 
  theme(axis.title.y = element_blank()) + 
  theme(title = element_text(hjust = 0.5, size = 30)) + 
  theme(plot.title = element_text(hjust = 0.6)) + 
  theme(axis.title.y = element_blank()) + 
  theme(axis.text.y = element_text(angle = 40))
```

## Graph all allele and Genotype Frequencies pre- and post-filtering
```{r}
prefilter <- plot_grid(ref_histogram, alt_histogram, homo_ref_histogram, homo_alt_histogram, het_histogram, nrow = 5, labels = c("a","b","c","d","e"), label_size = 30, label_x = 0.001, label_y = 1)

postfilter <- plot_grid(ref_histogram_postfilter, alt_histogram_postfilter, homo_ref_histogram_postfilter, homo_alt_histogram_postfilter, het_histogram_postfilter, nrow = 5, labels = c("f","g","h","i","j"), label_size = 30, label_x = 0.001, label_y = 1)

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Smallmouth_Bass_Genomics/visualization/preliminary_filtering_figures/allele_freq_plots.pdf", width=15, height=20)

plot_grid(prefilter, postfilter, ncol=2)

dev.off()

```



