---
title: "SMB_FINERADSTRUCTURE"
author: "Joe Gunn"
date: "11/12/2019"
output: html_document
---

# SMB Fineradstructure Population Discovery Analysis

## Libraries needed for analysis
```{r libraries, echo = FALSE, include=FALSE}
library(readxl)
library(tidyverse)
library(cowplot)
library(devtools)
library(SNPRelate)
library(vcfR)
library(pophelper)
library(PopGenome)
library(adegenet)
library(XML)
library(RColorBrewer)
source("../fineradstructure_analysis/FinestructureLibrary.R") #Fine-structure R source code
```

# FineRADstructure Analyses
### haplotype files were produced using the hapsFromVCF on my vcf file, and coancestry matrics were produced using the RADpainter and finestructure commands. R code and figures were adapted from the fineRADstructurePlot.R and FinestructureLibrary.R libraries (Milan Malinski)
## Read in data output from fineRADstructure 
```{r}
##All Samples 

#Finerad files for All SAMPLES, including Smallmouth and Spotted Bass 
chunkfile_blackbass <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_blackbass_chunks.out"
mcmcfile_blackbass <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_blackbass_chunks.mcmc.xml"
treefile_blackbass <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_blackbass_chunks.mcmcTree.xml"

#All samples, without the Spotted Bass Hybrid
chunkfile_nohybrid <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_nohybrid_chunks.out"
mcmcfile_nohybrid <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_nohybrid_chunks.mcmc.xml"
treefile_nohybrid <- "../../raw_data/fineradstructure_data/all_samples/hap_withfilters_nohybrid_chunks.mcmcTree.xml"

##Pure samples

#Finerad files for Only Non-Admixed Samples, including Spotted Bass
chunkfile_noadmix <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_chunks.out"
mcmcfile_noadmix <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_chunks.mcmc.xml"
treefile_noadmix <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_chunks.mcmcTree.xml"

#Finerad files for Only Non-Admixed SMB Samples, excluding Spotted Bass
chunkfile_noadmix_smb <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_smb_chunks.out"
mcmcfile_noadmix_smb <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_smb_chunks.mcmc.xml"
treefile_noadmix_smb <- "../../raw_data/fineradstructure_data/pure_samples/hap_withfilters_noadmix_smb_chunks.mcmcTree.xml"

##Admixed samples

#Finerad files for Only Admixed SMB Samples
chunkfile_admix <- "../../raw_data/fineradstructure_data/admixed_samples/hap_withfilters_admixed_chunks.out"
mcmcfile_admix <- "../../raw_data/fineradstructure_data/admixed_samples/hap_withfilters_admixed_chunks.mcmc.xml"
treefile_admix <- "../../raw_data/fineradstructure_data/admixed_samples/hap_withfilters_admixed_chunks.mcmcTree.xml"

##Hierarchical Structure 

#Finerad files for SMB SAMPLES, including Neosho and Nortern SMB
chunkfile_smb <- "../../raw_data/fineradstructure_data/smb_samples/hap_withfilters_smb_chunks.out"
mcmcfile_smb <- "../../raw_data/fineradstructure_data/smb_samples/hap_withfilters_smb_chunks.mcmc.xml"
treefile_smb <- "../../raw_data/fineradstructure_data/smb_samples/hap_withfilters_smb_chunks.mcmcTree.xml"

#Finerad files for Neosho SMB samples only
chunkfile_neo <- "../../raw_data/fineradstructure_data/neosho_samples/hap_withfilters_neosho_chunks.out"
mcmcfile_neo <- "../../raw_data/fineradstructure_data/neosho_samples/hap_withfilters_neosho_chunks.mcmc.xml"
treefile_neo <- "../../raw_data/fineradstructure_data/neosho_samples/hap_withfilters_neosho_chunks.mcmcTree.xml"

#Finerad files for Northern SMB samples only
chunkfile_nor <- "../../raw_data/fineradstructure_data/northern_samples/hap_withfilters_northern_chunks.out"
mcmcfile_nor <- "../../raw_data/fineradstructure_data/northern_samples/hap_withfilters_northern_chunks.mcmc.xml"
treefile_nor <- "../../raw_data/fineradstructure_data/northern_samples/hap_withfilters_northern_chunks.mcmcTree.xml"

#Finerad files for Neosho Non-Arkansas (no Leemulb or Bayou) SMB samples only
chunkfile_nonark <- "../../raw_data/fineradstructure_data/nonark_samples/hap_withfilters_nonark_chunks.out"
mcmcfile_nonark <- "../../raw_data/fineradstructure_data/nonark_samples/hap_withfilters_nonark_chunks.mcmc.xml"
treefile_nonark <- "../../raw_data/fineradstructure_data/nonark_samples/hap_withfilters_nonark_chunks.mcmcTree.xml"
```

## Set path name for depositing completed fineRADstructure figures
```{r}
### 2) EDIT THIS PATH TO WHERE YOU WANT THE PLOTS:
plotsFolder <- "../../visualization/fineradstructure_figures"

### 3) SET VALUES FOR THESE VARIABLES: "analysisName" will be included in output plots
analysisName <- "smb_coancestry" 
maxIndv <- 10000
maxPop <-10000
```

## Prepare color palette 
```{r color palette, echo=FALSE}
my_palette <- colorRampPalette(c("white", "navyblue","deeppink2","black"))(n = 299)
```

## Prepare Plots for ALL SAMPLES

### I am  running the coancestry matrix for all samples, excluding spotted bass (sister species) and excluding the hybrid from ADMIXTURE analysis (see SMB_ADMIXTURE.Rmd)

## Prepare raw coancestry data and tree
```{r}
#All ADMIXED Smallmouth Bass (SMB) samples
dataraw_admix <- as.matrix(read.table(chunkfile_admix, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_admix <- xmlTreeParse(mcmcfile_admix) ## read into xml format
mcmcdata_admix <- as.data.frame.myres(mcmcxml_admix) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_admix <- xmlTreeParse(treefile_admix) ## read the tree as xml format
ttree_admix <- extractTree(treexml_admix) ## extract the tree into ape's phylo format


#All SMB Smallmouth Bass samples
dataraw_smb <- as.matrix(read.table(chunkfile_smb, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_smb <- xmlTreeParse(mcmcfile_smb) ## read into xml format
mcmcdata_smb <- as.data.frame.myres(mcmcxml_smb) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_smb <- xmlTreeParse(treefile_smb) ## read the tree as xml format
ttree_smb <- extractTree(treexml_smb) ## extract the tree into ape's phylo format

#All pure Smallmouth Bass (SMB) samples
dataraw_noadmix_smb <- as.matrix(read.table(chunkfile_noadmix_smb, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_noadmix_smb <- xmlTreeParse(mcmcfile_noadmix_smb) ## read into xml format
mcmcdata_noadmix_smb <- as.data.frame.myres(mcmcxml_noadmix_smb) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_noadmix_smb <- xmlTreeParse(treefile_noadmix_smb) ## read the tree as xml format
ttree_noadmix_smb <- extractTree(treexml_noadmix_smb) ## extract the tree into ape's phylo format

#All pure Black Bass samples
dataraw_noadmix <- as.matrix(read.table(chunkfile_noadmix, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_noadmix <- xmlTreeParse(mcmcfile_noadmix) ## read into xml format
mcmcdata_noadmix <- as.data.frame.myres(mcmcxml_noadmix) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_noadmix <- xmlTreeParse(treefile_noadmix) ## read the tree as xml format
ttree_noadmix <- extractTree(treexml_noadmix) ## extract the tree into ape's phylo format

#All Black Bass Samples, without the SPOTTED BASS HYBRID
dataraw_nohybrid <- as.matrix(read.table(chunkfile_nohybrid, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_nohybrid <- xmlTreeParse(mcmcfile_nohybrid) ## read into xml format
mcmcdata_nohybrid <- as.data.frame.myres(mcmcxml_nohybrid) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_nohybrid <- xmlTreeParse(treefile_nohybrid) ## read the tree as xml format
ttree_nohybrid <- extractTree(treexml_nohybrid) ## extract the tree into ape's phylo format


#All Black Bass Samples
dataraw_blackbass <- as.matrix(read.table(chunkfile_blackbass, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_blackbass <- xmlTreeParse(mcmcfile_blackbass) ## read into xml format
mcmcdata_blackbass <- as.data.frame.myres(mcmcxml_blackbass) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_blackbass <- xmlTreeParse(treefile_blackbass) ## read the tree as xml format
ttree_blackbass <- extractTree(treexml_blackbass) ## extract the tree into ape's phylo format


#Neosho Samples
dataraw_neo <- as.matrix(read.table(chunkfile_neo, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_neo <- xmlTreeParse(mcmcfile_neo) ## read into xml format
mcmcdata_neo <- as.data.frame.myres(mcmcxml_neo) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_neo <- xmlTreeParse(treefile_neo) ## read the tree as xml format
ttree_neo <- extractTree(treexml_neo) ## extract the tree into ape's phylo format

#Northern Samples
dataraw_nor <- as.matrix(read.table(chunkfile_nor, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_nor <- xmlTreeParse(mcmcfile_nor) ## read into xml format
mcmcdata_nor <- as.data.frame.myres(mcmcxml_nor) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_nor <- xmlTreeParse(treefile_nor) ## read the tree as xml format
ttree_nor <- extractTree(treexml_nor) ## extract the tree into ape's phylo format

#Non-ark Samples
dataraw_nonark <- as.matrix(read.table(chunkfile_nonark, row.names=1, header=T, skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml_nonark <- xmlTreeParse(mcmcfile_nonark) ## read into xml format
mcmcdata_nonark <- as.data.frame.myres(mcmcxml_nonark) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml_nonark <- xmlTreeParse(treefile_nonark) ## read the tree as xml format
ttree_nonark <- extractTree(treexml_nonark) ## extract the tree into ape's phylo format
```

## Make files for coancestry plot and dendrogram
```{r}

#Only ADMIXED SMB samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_admix$node.label[ttree_admix$node.label!=""] <- format(as.numeric(ttree_admix$node.label[ttree_admix$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_admix <- myapetodend(ttree_admix, factor=1)
## Now we work on the MAP state
mapstate_admix <- extractValue(treexml_admix,"Pop") # map state as a finestructure clustering
mapstatelist_admix <- popAsList(mapstate_admix) # .. and as a list of individuals in populations
popnames_admix <- lapply(mapstatelist_admix, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_admix <-lapply(mapstatelist_admix, NameMoreSummary) # a nicer summary of the populations

names(popnames_admix) <- popnamesplot_admix # for nicety only
names(popnamesplot_admix ) <- popnamesplot_admix # for nicety only

popdend_admix <- makemydend(tdend_admix, mapstatelist_admix) # use NameSummary to make popdend
popdend_admix <- fixMidpointMembers(popdend_admix) # needed for obscure dendrogram reasons
popdendclear_admix <- makemydend(tdend_admix, mapstatelist_admix, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_admix <- fixMidpointMembers(popdendclear_admix) # needed for obscure dendrogram reasons


#Only SMB samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_smb$node.label[ttree_smb$node.label!=""] <- format(as.numeric(ttree_smb$node.label[ttree_smb$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_smb <- myapetodend(ttree_smb, factor=1)
## Now we work on the MAP state
mapstate_smb <- extractValue(treexml_smb,"Pop") # map state as a finestructure clustering
mapstatelist_smb <- popAsList(mapstate_smb) # .. and as a list of individuals in populations
popnames_smb <- lapply(mapstatelist_smb, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_smb <-lapply(mapstatelist_smb, NameMoreSummary) # a nicer summary of the populations

names(popnames_smb) <- popnamesplot_smb # for nicety only
names(popnamesplot_smb ) <- popnamesplot_smb # for nicety only

popdend_smb <- makemydend(tdend_smb, mapstatelist_smb) # use NameSummary to make popdend
popdend_smb <- fixMidpointMembers(popdend_smb) # needed for obscure dendrogram reasons
popdendclear_smb <- makemydend(tdend_smb, mapstatelist_smb, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_smb <- fixMidpointMembers(popdendclear_smb) # needed for obscure dendrogram reasons



#Only Pure SMB samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_noadmix_smb$node.label[ttree_noadmix_smb$node.label!=""] <- format(as.numeric(ttree_noadmix_smb$node.label[ttree_noadmix_smb$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_noadmix_smb <- myapetodend(ttree_noadmix_smb, factor=1)
## Now we work on the MAP state
mapstate_noadmix_smb <- extractValue(treexml_noadmix_smb,"Pop") # map state as a finestructure clustering
mapstatelist_noadmix_smb <- popAsList(mapstate_noadmix_smb) # .. and as a list of individuals in populations
popnames_noadmix_smb <- lapply(mapstatelist_noadmix_smb, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_noadmix_smb <-lapply(mapstatelist_noadmix_smb, NameMoreSummary) # a nicer summary of the populations

names(popnames_noadmix_smb) <- popnamesplot_noadmix_smb # for nicety only
names(popnamesplot_noadmix_smb ) <- popnamesplot_noadmix_smb # for nicety only

popdend_noadmix_smb <- makemydend(tdend_noadmix_smb, mapstatelist_noadmix_smb) # use NameSummary to make popdend
popdend_noadmix_smb <- fixMidpointMembers(popdend_noadmix_smb) # needed for obscure dendrogram reasons
popdendclear_noadmix_smb <- makemydend(tdend_noadmix_smb, mapstatelist_noadmix_smb, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_noadmix_smb <- fixMidpointMembers(popdendclear_noadmix_smb) # needed for obscure dendrogram reasons


#Only Pure samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_noadmix$node.label[ttree_noadmix$node.label!=""] <- format(as.numeric(ttree_noadmix$node.label[ttree_noadmix$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_noadmix <- myapetodend(ttree_noadmix, factor=1)
## Now we work on the MAP state
mapstate_noadmix <- extractValue(treexml_noadmix,"Pop") # map state as a finestructure clustering
mapstatelist_noadmix <- popAsList(mapstate_noadmix) # .. and as a list of individuals in populations
popnames_noadmix <- lapply(mapstatelist_noadmix, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_noadmix <-lapply(mapstatelist_noadmix, NameMoreSummary) # a nicer summary of the populations

names(popnames_noadmix) <- popnamesplot_noadmix # for nicety only
names(popnamesplot_noadmix ) <- popnamesplot_noadmix # for nicety only

popdend_noadmix <- makemydend(tdend_noadmix, mapstatelist_noadmix) # use NameSummary to make popdend
popdend_noadmix <- fixMidpointMembers(popdend_noadmix) # needed for obscure dendrogram reasons
popdendclear_noadmix <- makemydend(tdend_noadmix, mapstatelist_noadmix, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_noadmix <- fixMidpointMembers(popdendclear_noadmix) # needed for obscure dendrogram reasons



#Black Bass Samples, excluding the SPOTTED BASS HYBRID

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_nohybrid$node.label[ttree_nohybrid$node.label!=""] <- format(as.numeric(ttree_nohybrid$node.label[ttree_nohybrid$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_nohybrid <- myapetodend(ttree_nohybrid, factor=1)
## Now we work on the MAP state
mapstate_nohybrid <- extractValue(treexml_nohybrid,"Pop") # map state as a finestructure clustering
mapstatelist_nohybrid <- popAsList(mapstate_nohybrid) # .. and as a list of individuals in populations
popnames_nohybrid <- lapply(mapstatelist_nohybrid, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_nohybrid <-lapply(mapstatelist_nohybrid, NameMoreSummary) # a nicer summary of the populations

names(popnames_nohybrid) <- popnamesplot_nohybrid # for nicety only
names(popnamesplot_nohybrid ) <- popnamesplot_nohybrid # for nicety only

popdend_nohybrid <- makemydend(tdend_nohybrid, mapstatelist_nohybrid) # use NameSummary to make popdend
popdend_nohybrid <- fixMidpointMembers(popdend_nohybrid) # needed for obscure dendrogram reasons
popdendclear_nohybrid <- makemydend(tdend_nohybrid, mapstatelist_nohybrid, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_nohybrid <- fixMidpointMembers(popdendclear_nohybrid) # needed for obscure dendrogram reasons

#Black Bass Samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_blackbass$node.label[ttree_blackbass$node.label!=""] <- format(as.numeric(ttree_blackbass$node.label[ttree_blackbass$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_blackbass <- myapetodend(ttree_blackbass, factor=1)
## Now we work on the MAP state
mapstate_blackbass <- extractValue(treexml_blackbass,"Pop") # map state as a finestructure clustering
mapstatelist_blackbass <- popAsList(mapstate_blackbass) # .. and as a list of individuals in populations
popnames_blackbass <- lapply(mapstatelist_blackbass, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_blackbass <-lapply(mapstatelist_blackbass, NameMoreSummary) # a nicer summary of the populations

names(popnames_blackbass) <- popnamesplot_blackbass # for nicety only
names(popnamesplot_blackbass ) <- popnamesplot_blackbass # for nicety only

popdend_blackbass <- makemydend(tdend_blackbass, mapstatelist_blackbass) # use NameSummary to make popdend
popdend_blackbass <- fixMidpointMembers(popdend_blackbass) # needed for obscure dendrogram reasons
popdendclear_blackbass <- makemydend(tdend_blackbass, mapstatelist_blackbass, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_blackbass <- fixMidpointMembers(popdendclear_blackbass) # needed for obscure dendrogram reasons


#Neosho Samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_neo$node.label[ttree_neo$node.label!=""] <- format(as.numeric(ttree_neo$node.label[ttree_neo$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_neo <- myapetodend(ttree_neo, factor=1)
## Now we work on the MAP state
mapstate_neo <- extractValue(treexml_neo,"Pop") # map state as a finestructure clustering
mapstatelist_neo <- popAsList(mapstate_neo) # .. and as a list of individuals in populations
popnames_neo <- lapply(mapstatelist_neo, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_neo <-lapply(mapstatelist_neo, NameMoreSummary) # a nicer summary of the populations

names(popnames_neo) <- popnamesplot_neo # for nicety only
names(popnamesplot_neo ) <- popnamesplot_neo # for nicety only

popdend_neo <- makemydend(tdend_neo, mapstatelist_neo) # use NameSummary to make popdend
popdend_neo <- fixMidpointMembers(popdend_neo) # needed for obscure dendrogram reasons
popdendclear_neo <- makemydend(tdend_neo, mapstatelist_neo, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_neo <- fixMidpointMembers(popdendclear_neo) # needed for obscure dendrogram reasons

#Northern Samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_nor$node.label[ttree_nor$node.label!=""] <- format(as.numeric(ttree_nor$node.label[ttree_nor$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_nor <- myapetodend(ttree_nor, factor=1)
## Now we work on the MAP state
mapstate_nor <- extractValue(treexml_nor,"Pop") # map state as a finestructure clustering
mapstatelist_nor <- popAsList(mapstate_nor) # .. and as a list of individuals in populations
popnames_nor <- lapply(mapstatelist_nor, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_nor <-lapply(mapstatelist_nor, NameMoreSummary) # a nicer summary of the populations

names(popnames_nor) <- popnamesplot_nor # for nicety only
names(popnamesplot_nor ) <- popnamesplot_nor # for nicety only

popdend_nor <- makemydend(tdend_nor, mapstatelist_nor) # use NameSummary to make popdend
popdend_nor <- fixMidpointMembers(popdend_nor) # needed for obscure dendrogram reasons
popdendclear_nor <- makemydend(tdend_nor, mapstatelist_nor, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_nor <- fixMidpointMembers(popdendclear_nor) # needed for obscure dendrogram reasons

##Non-ark samples

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree_nonark$node.label[ttree_nonark$node.label!=""] <- format(as.numeric(ttree_nonark$node.label[ttree_nonark$node.label!=""]),digits=2)

 # convert to dendrogram format
tdend_nonark <- myapetodend(ttree_nonark, factor=1)
## Now we work on the MAP state
mapstate_nonark <- extractValue(treexml_nonark,"Pop") # map state as a finestructure clustering
mapstatelist_nonark <- popAsList(mapstate_nonark) # .. and as a list of individuals in populations
popnames_nonark <- lapply(mapstatelist_nonark, NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot_nonark <-lapply(mapstatelist_nonark, NameMoreSummary) # a nicer summary of the populations

names(popnames_nonark) <- popnamesplot_nonark # for nicety only
names(popnamesplot_nonark ) <- popnamesplot_nonark # for nicety only

popdend_nonark <- makemydend(tdend_nonark, mapstatelist_nonark) # use NameSummary to make popdend
popdend_nonark <- fixMidpointMembers(popdend_nonark) # needed for obscure dendrogram reasons
popdendclear_nonark <- makemydend(tdend_nonark, mapstatelist_nonark, "NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear_nonark <- fixMidpointMembers(popdendclear_nonark) # needed for obscure dendrogram reasons
```

## Print fineRADstructure Plots
```{r}
## Plot 1: COANCESTRY MATRIX
fullorder_admix <- labels(tdend_admix) # the order according to the tree
datamatrix_admix <- dataraw_admix[fullorder_admix, fullorder_admix] # reorder the data matrix

tmpmat_admix <- datamatrix_admix
tmpmat_admix[tmpmat_admix > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/admix_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_admix, 
                  dimnames(tmpmat_admix)[[1]], 
                  dend = tdend_admix,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_admix <- getPopMeanMatrix(datamatrix_admix, mapstatelist_admix)

tmpmat_admix <- popmeanmatrix_admix
tmpmat_admix[tmpmat_admix > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/admix_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_admix, 
                  dimnames(tmpmat_admix)[[1]], 
                  dend = tdend_admix, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_admix <- NameExpand(labels(popdend_admix))
mappopsizes_admix <- sapply(mappopcorrectorder_admix,length)
labellocs_admix <- PopCenters(mappopsizes_admix)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/admix_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_admix, 
                  dimnames(tmpmat_admix)[[1]], 
                  labelsx=labels(popdendclear_admix), 
                  labelsatx = labellocs_admix, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_admix, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()


## Plot 1: COANCESTRY MATRIX
fullorder_smb <- labels(tdend_smb) # the order according to the tree
datamatrix_smb <- dataraw_smb[fullorder_smb, fullorder_smb] # reorder the data matrix

tmpmat_smb <- datamatrix_smb
tmpmat_smb[tmpmat_smb > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/smb_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_smb, 
                  dimnames(tmpmat_smb)[[1]], 
                  dend = tdend_smb,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_smb <- getPopMeanMatrix(datamatrix_smb, mapstatelist_smb)

tmpmat_smb <- popmeanmatrix_smb
tmpmat_smb[tmpmat_smb > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/smb_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_smb, 
                  dimnames(tmpmat_smb)[[1]], 
                  dend = tdend_smb, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_smb <- NameExpand(labels(popdend_smb))
mappopsizes_smb <- sapply(mappopcorrectorder_smb,length)
labellocs_smb <- PopCenters(mappopsizes_smb)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/smb_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_smb, 
                  dimnames(tmpmat_smb)[[1]], 
                  labelsx=labels(popdendclear_smb), 
                  labelsatx = labellocs_smb, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_smb, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()



## Plot 1: COANCESTRY MATRIX
fullorder_noadmix_smb <- labels(tdend_noadmix_smb) # the order according to the tree
datamatrix_noadmix_smb <- dataraw_noadmix_smb[fullorder_noadmix_smb, fullorder_noadmix_smb] # reorder the data matrix

tmpmat_noadmix_smb <- datamatrix_noadmix_smb
tmpmat_noadmix_smb[tmpmat_noadmix_smb > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/noadmix_smb_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_noadmix_smb, 
                  dimnames(tmpmat_noadmix_smb)[[1]], 
                  dend = tdend_noadmix_smb,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 1: COANCESTRY MATRIX
fullorder_noadmix_smb <- labels(tdend_noadmix_smb) # the order according to the tree
datamatrix_noadmix_smb <- dataraw_noadmix_smb[fullorder_noadmix_smb, fullorder_noadmix_smb] # reorder the data matrix

tmpmat_noadmix_smb <- datamatrix_noadmix_smb
tmpmat_noadmix_smb[tmpmat_noadmix_smb > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/noadmix_smb_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_noadmix_smb, 
                  dimnames(tmpmat_noadmix_smb)[[1]], 
                  dend = tdend_noadmix_smb,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_noadmix_smb <- getPopMeanMatrix(datamatrix_noadmix_smb, mapstatelist_noadmix_smb)

tmpmat_noadmix_smb <- popmeanmatrix_noadmix_smb
tmpmat_noadmix_smb[tmpmat_noadmix_smb > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/noadmix_smb_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_noadmix_smb, 
                  dimnames(tmpmat_noadmix_smb)[[1]], 
                  dend = tdend_noadmix_smb, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_noadmix_smb <- NameExpand(labels(popdend_noadmix_smb))
mappopsizes_noadmix_smb <- sapply(mappopcorrectorder_noadmix_smb,length)
labellocs_noadmix_smb <- PopCenters(mappopsizes_noadmix_smb)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/noadmix_smb_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_noadmix_smb, 
                  dimnames(tmpmat_noadmix_smb)[[1]], 
                  labelsx=labels(popdendclear_noadmix_smb), 
                  labelsatx = labellocs_noadmix_smb, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_noadmix_smb, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()


## Plot 1: COANCESTRY MATRIX
fullorder_noadmix <- labels(tdend_noadmix) # the order according to the tree
datamatrix_noadmix <- dataraw_noadmix[fullorder_noadmix, fullorder_noadmix] # reorder the data matrix

tmpmat_noadmix <- datamatrix_noadmix
tmpmat_noadmix[tmpmat_noadmix > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/noadmix_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_noadmix, 
                  dimnames(tmpmat_noadmix)[[1]], 
                  dend = tdend_noadmix,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_noadmix <- getPopMeanMatrix(datamatrix_noadmix, mapstatelist_noadmix)

tmpmat_noadmix <- popmeanmatrix_noadmix
tmpmat_noadmix[tmpmat_noadmix > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/noadmix_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_noadmix, 
                  dimnames(tmpmat_noadmix)[[1]], 
                  dend = tdend_noadmix, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_noadmix <- NameExpand(labels(popdend_noadmix))
mappopsizes_noadmix <- sapply(mappopcorrectorder_noadmix,length)
labellocs_noadmix <- PopCenters(mappopsizes_noadmix)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/noadmix_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_noadmix, 
                  dimnames(tmpmat_noadmix)[[1]], 
                  labelsx=labels(popdendclear_noadmix), 
                  labelsatx = labellocs_noadmix, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_noadmix, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()



## Plot 1: COANCESTRY MATRIX
fullorder_nohybrid <- labels(tdend_nohybrid) # the order according to the tree
datamatrix_nohybrid <- dataraw_nohybrid[fullorder_nohybrid, fullorder_nohybrid] # reorder the data matrix

tmpmat_nohybrid <- datamatrix_nohybrid
tmpmat_nohybrid[tmpmat_nohybrid > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/nohybrid_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_nohybrid, 
                  dimnames(tmpmat_nohybrid)[[1]], 
                  dend = tdend_nohybrid,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_nohybrid <- getPopMeanMatrix(datamatrix_nohybrid, mapstatelist_nohybrid)

tmpmat_nohybrid <- popmeanmatrix_nohybrid
tmpmat_nohybrid[tmpmat_nohybrid > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nohybrid_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_nohybrid, 
                  dimnames(tmpmat_nohybrid)[[1]], 
                  dend = tdend_nohybrid, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_nohybrid <- NameExpand(labels(popdend_nohybrid))
mappopsizes_nohybrid <- sapply(mappopcorrectorder_nohybrid,length)
labellocs_nohybrid <- PopCenters(mappopsizes_nohybrid)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nohybrid_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_nohybrid, 
                  dimnames(tmpmat_nohybrid)[[1]], 
                  labelsx=labels(popdendclear_nohybrid), 
                  labelsatx = labellocs_nohybrid, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_nohybrid, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()


## Plot 1: COANCESTRY MATRIX
fullorder_blackbass <- labels(tdend_blackbass) # the order according to the tree
datamatrix_blackbass <- dataraw_blackbass[fullorder_blackbass, fullorder_blackbass] # reorder the data matrix

tmpmat_blackbass <- datamatrix_blackbass
tmpmat_blackbass[tmpmat_blackbass > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/blackbass_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_blackbass, 
                  dimnames(tmpmat_blackbass)[[1]], 
                  dend = tdend_blackbass,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_blackbass <- getPopMeanMatrix(datamatrix_blackbass, mapstatelist_blackbass)

tmpmat_blackbass <- popmeanmatrix_blackbass
tmpmat_blackbass[tmpmat_blackbass > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/blackbass_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_blackbass, 
                  dimnames(tmpmat_blackbass)[[1]], 
                  dend = tdend_blackbass, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_blackbass <- NameExpand(labels(popdend_blackbass))
mappopsizes_blackbass <- sapply(mappopcorrectorder_blackbass,length)
labellocs_blackbass <- PopCenters(mappopsizes_blackbass)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/blackbass_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_blackbass, 
                  dimnames(tmpmat_blackbass)[[1]], 
                  labelsx=labels(popdendclear_blackbass), 
                  labelsatx = labellocs_blackbass, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_blackbass, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()

##################

## Plot 1: COANCESTRY MATRIX
fullorder_neo <- labels(tdend_neo) # the order according to the tree
datamatrix_neo <- dataraw_neo[fullorder_neo, fullorder_neo] # reorder the data matrix

tmpmat_neo <- datamatrix_neo
tmpmat_neo[tmpmat_neo > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/neo_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_neo, 
                  dimnames(tmpmat_neo)[[1]], 
                  dend = tdend_neo,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_neo <- getPopMeanMatrix(datamatrix_neo, mapstatelist_neo)

tmpmat_neo <- popmeanmatrix_neo
tmpmat_neo[tmpmat_neo > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/neo_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_neo, 
                  dimnames(tmpmat_neo)[[1]], 
                  dend = tdend_neo, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_neo <- NameExpand(labels(popdend_neo))
mappopsizes_neo <- sapply(mappopcorrectorder_neo,length)
labellocs_neo <- PopCenters(mappopsizes_neo)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/neo_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_neo, 
                  dimnames(tmpmat_neo)[[1]], 
                  labelsx=labels(popdendclear_neo), 
                  labelsatx = labellocs_neo, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_neo, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()

##################

## Plot 1: COANCESTRY MATRIX
fullorder_nor <- labels(tdend_nor) # the order according to the tree
datamatrix_nor <- dataraw_nor[fullorder_nor, fullorder_nor] # reorder the data matrix

tmpmat_nor <- datamatrix_nor
tmpmat_nor[tmpmat_nor > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/nor_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_nor, 
                  dimnames(tmpmat_nor)[[1]], 
                  dend = tdend_nor,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_nor <- getPopMeanMatrix(datamatrix_nor, mapstatelist_nor)

tmpmat_nor <- popmeanmatrix_nor
tmpmat_nor[tmpmat_nor > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nor_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_nor, 
                  dimnames(tmpmat_nor)[[1]], 
                  dend = tdend_nor, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_nor <- NameExpand(labels(popdend_nor))
mappopsizes_nor <- sapply(mappopcorrectorder_nor,length)
labellocs_nor <- PopCenters(mappopsizes_nor)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nor_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_nor, 
                  dimnames(tmpmat_nor)[[1]], 
                  labelsx=labels(popdendclear_nor), 
                  labelsatx = labellocs_nor, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_nor, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()

##################

## Plot 1: COANCESTRY MATRIX
fullorder_nonark <- labels(tdend_nonark) # the order according to the tree
datamatrix_nonark <- dataraw_nonark[fullorder_nonark, fullorder_nonark] # reorder the data matrix

tmpmat_nonark <- datamatrix_nonark
tmpmat_nonark[tmpmat_nonark > maxIndv] <- maxIndv #  # cap the heatmap

pdf("../../visualization/fineradstructure_figures/final_figures/nonark_coancestry_matrix_betterlabels.pdf", height = 25, width = 25)

plotFinestructure(tmpmat_nonark, 
                  dimnames(tmpmat_nonark)[[1]], 
                  dend = tdend_nonark,
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2))

dev.off()


## Plot 2: POPULATIONS AND COANCESTRY AVERAGES
popmeanmatrix_nonark <- getPopMeanMatrix(datamatrix_nonark, mapstatelist_nonark)

tmpmat_nonark <- popmeanmatrix_nonark
tmpmat_nonark[tmpmat_nonark > maxPop] <- maxPop # cap the heatmap

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nonark_coancestry_matrix_averages_betterlabels.pdf", height = 20, width = 20)

plotFinestructure(tmpmat_nonark, 
                  dimnames(tmpmat_nonark)[[1]], 
                  dend = tdend_nonark, 
                  cols = my_palette, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                                 t.srt = 1, 
                                 t.off = -0.5, 
                                 t.cex = 2))

dev.off()


## Plot 3: POPULATIONS AND COANCESTRY AVERAGES WITH PERHAPS MORE INFORMATIVE LABELS
mappopcorrectorder_nonark <- NameExpand(labels(popdend_nonark))
mappopsizes_nonark <- sapply(mappopcorrectorder_nonark,length)
labellocs_nonark <- PopCenters(mappopsizes_nonark)
xcrt=0
ycrt=45

pdf("../../visualization/fineradstructure_figures/unused_intermediate_figures/nonark_coancestry_matrix_averages_betterlabels_betterlabels.pdf", height = 25, width = 25)
 
plotFinestructure(tmpmat_nonark, 
                  dimnames(tmpmat_nonark)[[1]], 
                  labelsx=labels(popdendclear_nonark), 
                  labelsatx = labellocs_nonark, 
                  xcrt = xcrt, 
                  cols = my_palette, 
                  ycrt = ycrt, 
                  dend = tdend_nonark, 
                  cex.axis = 1.5, 
                  edgePar = list(p.lwd = 0, 
                               t.srt = 1, 
                               t.off = -0.5, 
                               t.cex = 2), 
                  hmmar = c(3,0,0,1))

dev.off()
```